<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.28">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Ryan V. Crawford">
<meta name="author" content="Jamie Crawford">
<meta name="author" content="Julie L. Hansen">
<meta name="author" content="Lawrence B. Smart">
<meta name="author" content="Virginia Moore">
<meta name="dcterms.date" content="2024-04-03">
<meta name="keywords" content="Hemp, Grain, Spectroscopy">

<title>Near Infrared Spectroscopy Predicts Percentage Crude Protein in Hemp Grain</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<meta name="citation_title" content="Near Infrared Spectroscopy Predicts Percentage Crude Protein in Hemp Grain">
<meta name="citation_abstract" content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
">
<meta name="citation_keywords" content="Hemp,Grain,Spectroscopy">
<meta name="citation_author" content="Ryan V. Crawford">
<meta name="citation_author" content="Jamie Crawford">
<meta name="citation_author" content="Julie L. Hansen">
<meta name="citation_author" content="Lawrence B. Smart">
<meta name="citation_author" content="Virginia Moore">
<meta name="citation_publication_date" content="2024-04-03">
<meta name="citation_cover_date" content="2024-04-03">
<meta name="citation_year" content="2024">
<meta name="citation_online_date" content="2024-04-03">
<meta name="citation_language" content="en">
<meta name="citation_reference" content="citation_title=NIR Instruments and Prediction Methods for Rapid Access to Grain Protein Content in Multiple Cereals.;,citation_abstract=Achieving global goals for sustainable nutrition, health, and wellbeing will depend on delivering enhanced diets to humankind. This will require instantaneous access to information on food-source quality at key points of agri-food systems. Although laboratory analysis and benchtop NIR spectrometers are regularly used to quantify grain quality, these do not suit all end users, for example, stakeholders in decentralized agri-food chains that are typical in emerging economies. Therefore, we explored benchtop and portable NIR instruments, and the methods that might aid these particular end uses. For this purpose, we generated NIR spectra for 328 grain samples from multiple cereals (finger millet, foxtail millet, maize, pearl millet, and sorghum) with a standard benchtop NIR spectrometer (DS2500, FOSS) and a novel portable NIR-based instrument (HL-EVT5, Hone). We explored classical deterministic methods (via winISI, FOSS), novel machine learning (ML)-driven methods (via Hone Create, Hone), and a convolutional neural network (CNN)-based method for building the calibrations to predict grain protein out of the NIR spectra. All of the tested methods enabled us to build relevant calibrations out of both types of spectra (i.e., R ^{\textrm{2}} ≥ 0.90, RMSE ≤ 0.91, RPD ≥ 3.08). Generally, the calibration methods integrating the ML techniques tended to enhance the prediction capacity of the model. We also documented that the prediction of grain protein content based on the NIR spectra generated using the novel portable instrument (HL-EVT5, Hone) was highly relevant for quantitative protein predictions (R ^{\textrm{2}} = 0.91, RMSE = 0.97, RPD = 3.48). Thus, the presented findings lay the foundations for the expanded use of NIR spectroscopy in agricultural research, development, and trade.;,citation_author=Keerthi Chadalavada;,citation_author=Krithika Anbazhagan;,citation_author=Adama Ndour;,citation_author=Sunita Choudhary;,citation_author=William Palmer;,citation_author=Jamie R Flynn;,citation_author=Srikanth Mallayee;,citation_author=Sharada Pothu;,citation_author=Kodukula Venkata Subrahamanya Vara Prasad;,citation_author=Padmakumar Varijakshapanikar;,citation_author=Chris S Jones;,citation_author=Jana Kholová;,citation_publication_date=2022-05;,citation_cover_date=2022-05;,citation_year=2022;,citation_issue=10;,citation_doi=10.3390/s22103710;,citation_issn=1424-8220;,citation_volume=22;,citation_journal_title=Sensors (Basel, Switzerland);">
<meta name="citation_reference" content="citation_title=Estimation of crude protein and amino acid contents in whole, ground and defatted ground soybeans by different types of near-infrared (NIR) reflectance spectroscopy;,citation_abstract=This study was designed to examine the ability of near-infrared (NIR) spectroscopy to determine amino acid and crude protein concentrations in soybean and explore the effects of particle size, fat content and types of NIR spectroscopy instruments on the predictive ability of the NIR models. Whole, ground and defatted ground soybean were scanned by a diode array NIR analyzer and defatted ground soybean were scanned by a Fourier transformed NIR analyzer. Samples were analyzed for crude protein and amino acid contents using reference wet chemistry methods. The NIR calibration models of crude protein and most amino acids showed acceptable coefficient of determinations (R2c = 0.605–0.952), while the NIR models for Cys, Met and Trp were less accurate (R2c = 0.498–0.667 for Cys and R2c = 0.482–0.615 for Met and R2c = 0.406–0.481 for Trp). The R2c values in NIR calibration were found to relate to the correlation between amino acids and crude protein. The grinding process and lipid extraction were found to improve the NIR spectroscopy in crude protein and amino acids prediction. The Fourier transformed NIR analyzer was found to perform similar to the diode array NIR analyzer for defatted ground soybean analysis. No significant differences were found between crude protein and amino acid contents predicted by these two NIR spectroscopy instruments (P&nbsp;&amp;amp;amp;gt;&nbsp;0.05).;,citation_author=Da Shi;,citation_author=Jiayi Hang;,citation_author=Jason Neufeld;,citation_author=Shusheng Zhao;,citation_author=James D. House;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0889157522002198;,citation_doi=10.1016/j.jfca.2022.104601;,citation_issn=0889-1575;,citation_volume=111;,citation_journal_title=Journal of Food Composition and Analysis;">
<meta name="citation_reference" content="citation_title=Comparison of PLSR, MLR, SVM regression methods for determination of crude protein and carbohydrate content in stored wheat using near Infrared spectroscopy;,citation_abstract=Aim of the study was to predict quality parameters namely protein and carbohydrate content of wheat grain stored for one year at 4&nbsp;°C and 21&nbsp;°C using Near Infrared spectroscopy (NIRS) and Chemometrics. The spectra of grains were measured in reflectance mode using lab made filter based pre dispersive NIRS ranging from wavelength 750&nbsp;nm to 2580&nbsp;nm. NIR Wavelengths were divided into two sets for all the stored samples depending on its correlation with factors. The reference data and NIR data were analysed by principal component analysis (PCA) scores, partial last squared regression (PLSR) model, Multiple Regression Analysis (MLR) and Support Vector Machine Regression (SVMR). First four factors were sufficient to develop regression models for the above mentioned methods. Prediction performance was compared on the basis of the coefficient of correlation (R2) and Root Mean Square Error (RMSE) for the calibration and validation sets. R2 values for prediction of protein content from PLSR model were 0.955, 0.997 for 4&nbsp;°C Wavelength Set I and II and 0.903, 0.989 for 21&nbsp;°C Wavelength Set I and II. For the prediction of carbohydrate content 0.978, 0.951 and 0.946, 0.974 were the R2 values for the 4&nbsp;°C and 21&nbsp;°C wheat samples Set I and Set II respectively. MLR model also gave R2 values greater than 0.9 for all the samples. Wavelengths with high β correlation coefficients were defined. SVMR predicted the protein content and carbohydrate content with good accuracy. SVMR and PLS results. All the models developed showed good accuracy for successful prediction of protein and carbohydrate content of stored wheat grains with PLSR and SVMR giving the best results. This study showed that near infrared spectroscopy has potential to distinguish wheat grains stored under different conditions.;,citation_author=Uma Kamboj;,citation_author=Paramita Guha;,citation_author=Sunita Mishra;,citation_publication_date=2022-01;,citation_cover_date=2022-01;,citation_year=2022;,citation_doi=10.1016/j.matpr.2021.04.540;,citation_issn=2214-7853;,citation_volume=48;,citation_journal_title=Materials Today: Proceedings;">
<meta name="citation_reference" content="citation_title=Intelligent evaluation of free amino acid and crude protein content in raw peanut seed kernels using NIR spectroscopy paired with multivariable calibration;,citation_abstract=Given the nutritional importance of peanuts, this study examined the free amino acid (FAA) and crude protein (CP) content in raw peanut seeds. Near-infrared spectroscopy (NIRS) was employed in combination with variable selection algorithms after successful reference data analysis using colorimetric and Kjeldahl methods. Ensuing the application of partial least squares (PLS) as a full spectral model, the genetic algorithm (GA), bootstrapping soft shrinkage (BOSS), uninformative variable elimination (UVE), and random frog (RF) models were tested and assessed. A comparison of correlation coefficients of prediction (Rₚ), root mean square error of prediction (RMSEP), and residual predictive deviation (RPD) was performed to appraise the performance of the built models. Using RF-PLS, an unsurpassed outcome was achieved for FAA (Rₚ = 0.937, RPD = 3.38) and CP (Rₚ = 0.9261, RPD = 3.66). These findings demonstrated that NIR in combination with RF-PLS could be utilized for quantitative, rapid, and nondestructive prediction of FAA and CP in raw peanut seed samples.;,citation_author=Suleiman A. Haruna;,citation_author=Huanhuan Li;,citation_author=Wenya Wei;,citation_author=Wenhui Geng;,citation_author=Selorm Yao-Say Solomon Adade;,citation_author=Muhammad Zareef;,citation_author=Ngouana Moffo A. Ivane;,citation_author=Quansheng Chen;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_issue=31;,citation_doi=10.1039/d2ay00875k;,citation_issn=1759-9679;,citation_volume=14;,citation_journal_title=Analytical methods;">
<meta name="citation_reference" content="citation_title=High throughput phenotyping of Camelina sativa seeds for crude protein, total oil, and fatty acids profile by near infrared spectroscopy;,citation_abstract=Highlights •Near infrared spectroscopy (NIRS) is a fast, non-destructive method for high throughput phenotyping of camelina seed.•NIRS works well for evaluating camelina seed crude protein, total oil, and fatty acid profiles.•NIRS calibration equations for phenotyping agronomic traits of camelina seed will be useful for conducting genome wide association studies.;,citation_author=James V. Anderson;,citation_author=Alex Wittenberg;,citation_author=Hui Li;,citation_author=Marisol T. Berti;,citation_publication_date=2019-10;,citation_cover_date=2019-10;,citation_year=2019;,citation_doi=10.1016/j.indcrop.2019.04.075;,citation_issn=0926-6690;,citation_volume=137;,citation_journal_title=Industrial Crops &amp;amp;amp; Products;">
<meta name="citation_reference" content="citation_title=Accuracy of NIRS Instruments to Analyze Forage and Grain1;,citation_abstract=A study was made of the relative accuracy of three near infrared reflectance (NIR) instruments: a monochromator and two lower cost scanning filter instruments. All instruments used in this test were manufactured by Pacific Scientific, Silver Spring, MD. The monochromator was the 6100 with a scanning range of 1100 to 2498 nm. The 51A had a scanning range from 1596 to 1756 nm and 1988 to 2356 nm. The 102 scanned from 1902 to 2320 nm. Computer programs developed for the National NIR Forage Project were used to conduct the instrument comparisons tests. Agricultural products used in the experiment were hay, haylage, corn (Zea mays L.) silage and corn grain. Analyses included crude protein (CP), acid detergent fiber (ADF), Ca, P, K, M, and dry matter (DM). The 6100 generally had lower standard errors of validation than the scanning filter instruments. Although the six-filter 51A had more spectral information than the three-filter 102, it did not consistently out-perform the 102. There was no trend for one math treatment or series of wavelengths to be selected across instruments, products, or analyses. Regardless of the instrument spectral capabilities, the fraction of explained variance (R2) values for CP, ADF, and DM were higher than the R2s for the minerals. Even though the accuracy of the filter instruments was often below that of the scanning monochromator, the differences were small and would be of little significance for many agricultural applications.;,citation_author=John S. Shenk;,citation_author=Mark O. Westerhaus;,citation_publication_date=1985;,citation_cover_date=1985;,citation_year=1985;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.2135/cropsci1985.0011183X002500060054x;,citation_issue=6;,citation_doi=10.2135/cropsci1985.0011183X002500060054x;,citation_issn=1435-0653;,citation_volume=25;,citation_journal_title=Crop Science;">
<meta name="citation_reference" content="citation_title=Moisture effects on robustness of sorghum grain protein near‐infrared spectroscopy calibration;,citation_abstract=BACKGROUND AND OBJECTIVES: A near‐infrared (NIR) spectroscopy method was developed for rapid and nondestructive evaluation of protein content of intact sorghum grains. Effect of grain sample moisture variation on the robustness of protein calibration was investigated. FINDINGS: An initial NIR protein calibration model with sorghum grains in 7.28%–11.57% moisture content range with coefficient of determination (R²) of 0.94 and standard error of cross‐validation (SECV) of 0.41%, predicted protein content of an external validation set of different varieties with R²&nbsp;=&nbsp;0.90, root‐mean‐square error of prediction (RMSEP)&nbsp;=&nbsp;0.42% and bias&nbsp;=&nbsp;−0.02%. However, when grains with a wider range of moisture content (7.50%–17.75%) were used for validation, prediction errors increased with R²&nbsp;=&nbsp;0.72 RMSEP&nbsp;=&nbsp;0.87% and bias&nbsp;=&nbsp;−0.32%. Inclusion of grains with a wider moisture range to the calibration set improved the performance of the calibration model with a R²&nbsp;=&nbsp;0.83, RMSEP&nbsp;=&nbsp;0.67% with a bias of −0.04%. CONCLUSIONS: Variation of moisture content in grains affected the performance of the NIR protein calibration model. Likewise, inclusion of moisture variation in the calibration sample set improved the robustness of the model. SIGNIFICANCE AND NOVELTY: In addition to the traits of interest, variation of other physical or chemical traits should also be considered for inclusion into the calibration sample set to improve the robustness of NIR calibration models. This is especially important when NIR spectroscopy methods are developed for evaluation of breeding populations as the future grain samples from numerous crosses may be substantially diverse.;,citation_author=Kamaranga H. S. Peiris;,citation_author=Scott R. Bean;,citation_author=Anuj Chiluwal;,citation_author=Ramasamy Perumal;,citation_author=S. V. Krishna Jagadish;,citation_publication_date=2019-01;,citation_cover_date=2019-01;,citation_year=2019;,citation_issue=4 p.678-688;,citation_doi=10.1002/cche.10164;,citation_issn=00090352;,citation_volume=96;,citation_journal_title=Cereal chemistry;">
<meta name="citation_reference" content="citation_title=Potential of Near- and Mid-infrared Spectroscopy in Biofuel Production;,citation_abstract=For many decades, near-infrared spectroscopy (NIRS) has been used to determine the composition of animal feedstuffs and grains. More recently, mid-infrared spectroscopy (mid-IR) has also been examined for similar determinations. These spectroscopic methods offer the potential for rapid and accurate determination of organic constituents, such as fiber components and protein, of forages, by-products, and grains at reduced cost and greatly increased speed (minutes instead of hours or days). Because they are nonchemical in nature, they result in a large reduction (90% or more) in the chemical wastes associated with standard chemical-based assay methods. The same components of interest for biofuel production (cellulose, hemicellulose, lignin, starch, protein, oil, etc.) are those that have already been determined by NIRS/mid-IR for evaluating grains and animal feedstuffs. Therefore, these techniques would appear to be a natural match for evaluating feedstocks for biofuels, and the literature shows that efforts in this direction are being successfully tested and instituted. For this discussion, an overview of where such efforts are and the potential for NIRS/mid-IR in producing biofuels will be covered. For example, while there are similarities between the needs of the biofuels industry and the analysis of animal feedstuffs, there are also both practical and technical differences between the two that will likely impact how NIRS/mid-IR is developed for biofuels. As an example, grain analysis for protein is performed on a large scale by government agencies such as the Canadian Grain Commission and U.S. Grain Inspection Service, while at least in the United States, animal feedstuff analysis is performed by state or independent laboratories for individual farmers. For biofuels, this might well result in most analysis being performed by the large corporations converting the feedstocks to biofuels, as opposed to the individual producer having analysis performed at an independent laboratory. Similarly for animal feedstuffs, measurements of fiber (neutral detergent fiber or NDF, acid detergent fiber or ADF, and lignin) and protein are carried out. These fiber measurements often consist of more than one type of fiber component with some being computed by difference (hemicellulose = NDF – ADF) and are empirical at best. Whether such empirical estimates will be sufficient for assessing biofuels or whether new spectroscopic methods for directly measuring the components of interest (cellulose, etc.) will need to be developed is a question to be answered when components other than starch for ethanol or oil for biodiesel become common.;,citation_author=James B. Reeves;,citation_publication_date=2012-01;,citation_cover_date=2012-01;,citation_year=2012;,citation_fulltext_html_url=https://doi.org/10.1080/00103624.2012.641844;,citation_issue=1-2;,citation_doi=10.1080/00103624.2012.641844;,citation_issn=0010-3624;,citation_volume=43;,citation_journal_title=Communications in Soil Science and Plant Analysis;">
<meta name="citation_reference" content="citation_title=Near-infrared spectroscopy in agriculture;,citation_author=Craig A Roberts;,citation_author=Jerry Workman;,citation_author=James B Reeves;,citation_publication_date=2004;,citation_cover_date=2004;,citation_year=2004;">
<meta name="citation_reference" content="citation_title=A note on Mahalanobis and related distance measures in WinISI and The Unscrambler;,citation_abstract=In identifying spectral outliers in near infrared calibration it is common to use a distance measure that is related to Mahalanobis distance. However, different software packages tend to use different variants, which lead to a translation problem if more than one package is used. Here the relationships between squared Mahalanobis distance D2, the GH distance of WinISI, and the T2 and leverage (L) statistics of The Unscrambler are established as D2?=?T2 ≈ L???n ≈ GH???k, where n and k are the numbers of samples and variables, respectively, in the set of spectral data used to establish the distance measure. The implications for setting thresholds for outlier detection are discussed. On the way to this result the principal component scores from WinISI and The Unscrambler are compared. Both packages scale the scores for a component to have variances proportional to the contribution of that component to total variance, but the WinISI scores, unlike those from The Unscrambler, do not have mean zero.;,citation_author=A Garrido-Varo;,citation_author=J Garcia-Olmo;,citation_author=T Fearn;,citation_publication_date=2019-08;,citation_cover_date=2019-08;,citation_year=2019;,citation_fulltext_html_url=https://doi.org/10.1177/0967033519848296;,citation_issue=4;,citation_doi=10.1177/0967033519848296;,citation_issn=0967-0335;,citation_volume=27;,citation_journal_title=Journal of Near Infrared Spectroscopy;">
<meta name="citation_reference" content="citation_title=Application of near infrared reflectance spectroscopy to analysis of cereal grains and oilseeds;,citation_author=P. C. Williams;,citation_publication_date=1975-07;,citation_cover_date=1975-07;,citation_year=1975;,citation_issue=4 p.561-576;,citation_issn=0009-0352;,citation_volume=52;,citation_journal_title=Cereal chemistry;">
<meta name="citation_reference" content="citation_title=The impact of hempseed dehulling on chemical composition, structure properties and aromatic profile of hemp protein isolate;,citation_abstract=Hempseed protein has attracted increasing attention in both scientific and industrial field because of its high nutritional value. In the current study, the influence of dehulling process on the chemical composition, structure and functional properties of hempseed protein isolate (HPI) was determined by using SDS-PAGE, SEC-HPLC-UV/RI/MALS, CD spectroscopy and DSC. The result demonstrated that dehulling process significantly increased the extraction and protein recovery yields of HPI. Meanwhile, dehulled HPI also exhibited a higher protein purity and Arg/Lys ratio with improved color. No significant impact on protein composition and structure was observed upon seed dehulling, but HPI extracted from the dehulled seeds showed a reduced thermal stability as compared to non-dehulled HPI. The aromatic profile of HPI was determined by HS-SPME-GC-MS, and the data suggested that dehulled HPI accumulated more terpenes and less lipid oxidation volatiles than non-dehulled HPI. These findings indicated that dehulling process greatly enhances the extraction yield with higher protein content and more characteristic aroma of HPI. The current study for the first time provides valuable information about the influence of dehulling processing on the properties of HPI, which will facilitate further research and application of hempseed and its valued added ingredients in both academic and industry field.;,citation_author=Peiyi Shen;,citation_author=Zili Gao;,citation_author=Minwei Xu;,citation_author=Jae-Bom Ohm;,citation_author=Jiajia Rao;,citation_author=Bingcan Chen;,citation_publication_date=2020-09;,citation_cover_date=2020-09;,citation_year=2020;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0268005X20301338;,citation_doi=10.1016/j.foodhyd.2020.105889;,citation_issn=0268-005X;,citation_volume=106;,citation_journal_title=Food Hydrocolloids;">
<meta name="citation_reference" content="citation_title=Genotype x Environmental Effects on Yielding Ability and Seed Chemical Composition of Industrial Hemp (Cannabis sativa L.) Varieties Grown in North Dakota, USA;,citation_abstract=The influence of growing conditions in North Dakota on grain yield and seed composition was evaluated for industrial hemp varieties (Cannabis sativa L.) grown in North Dakota, USA. Average grain yield across varieties was 2138 kg ha−1. Crude protein content of undehulled industrial hemp flour from 10 varieties ranged from 32.7% to 35.9%; and oil content ranged from 24.3% to 28.1%. All industrial hemp flours were excellent natural sources of iron (46.7%), manganese (169.1%), copper (29.0%), zinc (28.2%), phosphorus (41.0%), and magnesium (33.7%) on a per serving basis (30 g per serving) for percent recommended daily intake (% RDI). Oils from most of the varieties differed significantly for the fatty acid (FA) profile. The n6/n3 ratio ranged from 3.2 to 5.1. Principal component analysis (PCA) suggested that CFX-2 and CRS-1 grew well in North Dakota based on grain yield, flour, and oil quality. In this study, the chemical composition of flour was not influenced by crop year; but crop year did effect the nutritional mineral composition. These results may have important implications for the use of industrial hemp in food applications.;,citation_author=Yang Lan;,citation_author=Fengchao Zha;,citation_author=Allen Peckrul;,citation_author=Bryan Hanson;,citation_author=Burton Johnson;,citation_author=Jiajia Rao;,citation_author=Bingcan Chen;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1002/aocs.12291;,citation_issue=12;,citation_doi=10.1002/aocs.12291;,citation_issn=1558-9331;,citation_volume=96;,citation_journal_title=Journal of the American Oil Chemists’ Society;">
<meta name="citation_reference" content="citation_title=Hempseed as a nutritional resource: An overview;,citation_abstract=The seed of Cannabis sativa L. has been an important source of nutrition for thousands of years in Old World cultures. Non-drug varieties of Cannabis, commonly referred to as hemp, have not been studied extensively for their nutritional potential in recent years, nor has hempseed been utilized to any great extent by the industrial processes and food markets that have developed during the 20th century. Technically a nut, hempseed typically contains over 30% oil and about 25% protein, with considerable amounts of dietary fiber, vitamins and minerals. Hempseed oil is over 80% in polyunsaturated fatty acids (PUFAs), and is an exceptionally rich source of the two essential fatty acids (EFAs) linoleic acid (18:2 omega-6) and alpha-linolenic acid (18:3 omega-3). The omega-6 to omega-3 ratio (n6/n3) in hempseed oil is normally between 2:1 and 3:1, which is considered to be optimal for human health. In addition, the biological metabolites of the two EFAs, gamma-linolenic acid (18:3 omega-6; “GLA”) and stearidonic acid (18:4 omega-3; “SDA”), are also present in hempseed oil. The two main proteins in hempseed are edestin and albumin. Both of these high-quality storage proteins are easily digested and contain nutritionally significant amounts of all essential amino acids. In addition, hempseed has exceptionally high levels of the amino acid arginine. Hempseed has been used to treat various disorders for thousands of years in traditional oriental medicine. Recent clinical trials have identified hempseed oil as a functional food, and animal feeding studies demonstrate the long-standing utility of hempseed as an important food resource.;,citation_author=J. C. Callaway;,citation_publication_date=2004-01;,citation_cover_date=2004-01;,citation_year=2004;,citation_fulltext_html_url=https://doi.org/10.1007/s10681-004-4811-6;,citation_issue=1;,citation_doi=10.1007/s10681-004-4811-6;,citation_issn=1573-5060;,citation_volume=140;,citation_journal_title=Euphytica;">
<meta name="citation_reference" content="citation_title=Proteomic Profiles of Whole Seeds, Hulls, and Dehulled Seeds of Two Industrial Hemp (Cannabis sativa L.) Cultivars;,citation_abstract=As a source of nutritionally important components, hemp seeds are often dehulled for consumption and food applications by removing the hard hulls, which increases their nutritional value. The hulls thus become waste, although they may contain valuable protein items, about which there is a lack of information. The present work is therefore aimed at evaluating the proteome of hemp (Cannabis sativa L.) at the whole-seed, dehulled seed, and hull levels. The evaluation was performed on two cultivars, Santhica 27 and Uso-31, using LC-MS/MS analysis. In total, 2833 protein groups (PGs) were identified, and their relative abundances were determined. A set of 88 PGs whose abundance exceeded 1000 ppm (MP88 set) was considered for further evaluation. The PGs of the MP88 set were divided into ten protein classes. Seed storage proteins were found to be the most abundant protein class: the averages of the cultivars were 65.5%, 71.3%, and 57.5% for whole seeds, dehulled seeds, and hulls, respectively. In particular, 11S globulins representing edestin (three PGs) were found, followed by 7S vicilin-like proteins (four PGs) and 2S albumins (two PGs). The storage 11S globulins in Santhica 27 and Uso-31 were found to have a higher relative abundance in the dehulled seed proteome (summing to 58.6 and 63.2%) than in the hull proteome (50.5 and 54%), respectively. The second most abundant class of proteins was oleosins, which are part of oil-body membranes. PGs belonging to metabolic proteins (e.g., energy metabolism, nucleic acid metabolism, and protein synthesis) and proteins related to the defence and stress responses were more abundant in the hulls than in the dehulled seeds. The hulls can, therefore, be an essential source of proteins, especially for medical and biotechnological applications. Proteomic analysis has proven to be a valuable tool for studying differences in the relative abundance of proteins between dehulled hemp seeds and their hulls among different cultivars.;,citation_author=Jan Bárta;,citation_author=Pavel Roudnický;,citation_author=Markéta Jarošová;,citation_author=Zbyněk Zdráhal;,citation_author=Adéla Stupková;,citation_author=Veronika Bártová;,citation_author=Zlatuše Krejčová;,citation_author=Jan Kyselka;,citation_author=Vladimír Filip;,citation_author=Václav Říha;,citation_author=František Lorenc;,citation_author=Jan Bedrníček;,citation_author=Pavel Smetana;,citation_publication_date=2024-01;,citation_cover_date=2024-01;,citation_year=2024;,citation_fulltext_html_url=https://www.mdpi.com/2223-7747/13/1/111;,citation_issue=1;,citation_doi=10.3390/plants13010111;,citation_issn=2223-7747;,citation_volume=13;,citation_journal_title=Plants;">
<meta name="citation_reference" content="citation_title=Genetics of the Inverse Relationship between Grain Yield and Grain Protein Content in Common Wheat;,citation_abstract=Grain protein content (GPC) is one of the most important criteria to determine the quality of common wheat (Triticum aestivum). One of the major obstacles for bread wheat production is the negative correlation between GPC and grain yield (GY). Previous studies demonstrated that the deviation from this inverse relationship is highly heritable. However, little is known about the genetics controlling these deviations in common wheat. To fill this gap, we performed quantitative trait locus (QTL) analysis for GY, GPC, and four derived GY-GPC indices using an eight-way multiparent advanced generation intercross population comprising 394 lines. Interval mapping was conducted using phenotypic data from up to nine environments and genotypic data from a 20k single-nucleotide polymorphism array. The four indices were highly heritable (0.76–0.88) and showed distinct correlations to GY and GPC. Interval mapping revealed that GY, GPC, and GY-GPC indices were controlled by 6, 12, and 12 unique QTL, of which each explained only a small amount of phenotypic variance (R2 ≤ 10%). Ten of the 12 index QTL were independent of loci affecting GY and GPC. QTL regions harboured several candidate genes, including Rht-1, WAPO-A1, TaTEF-7A, and NRT2.6-7A. The study confirmed the usefulness of indices to mitigate the inverse GY-GPC relationship in breeding, though the selection method should reflect their polygenic inheritance.;,citation_author=Manuel Geyer;,citation_author=Volker Mohler;,citation_author=Lorenz Hartl;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9413592/;,citation_issue=16;,citation_doi=10.3390/plants11162146;,citation_issn=2223-7747;,citation_pmid=36015449;,citation_volume=11;,citation_journal_title=Plants;">
<meta name="citation_reference" content="citation_title=Genetic Variation for Protein Content and Yield-Related Traits in a Durum Population Derived From an Inter-Specific Cross Between Hexaploid and Tetraploid Wheat Cultivars;,citation_abstract=&amp;amp;amp;lt;p&amp;gt;Wheat grain protein content (GPC) and yield components are complex quantitative traits influenced by a multi-factorial system consisting of both genetic and environmental factors. Although seed storage proteins represent less than 15% of mature kernels, they are crucial in determining end-use properties of wheat, as well as the nutritional value of derived products. Yield and GPC are negatively correlated, and this hampers breeding programs of commercially valuable wheat varieties. The goal of this work was the evaluation of genetic variability for quantity and composition of seed storage proteins, together with yield components [grain yield per spike (GYS) and thousand-kernel weight (TKW)] in a durum wheat population obtained by an inter-specific cross between a common wheat accession and the durum cv. Saragolla. Quantitative trait loci (QTL) analysis was conducted and closely associated markers identified on a genetic map composed of 4,366 SNP markers previously obtained in the same durum population genotyped with the 90K iSelect SNP assay. A total of 22 QTL were detected for traits related to durum wheat quality. Six genomic regions responsible for GPC control were mapped on chromosomes 2B, 3A, 4A, 4B, 5B, and 7B, with major QTL on chromosomes 2B, 4A, and 5B. Nine loci were detected for GYS: two on chromosome 5B and 7A and one on chromosomes 2A, 2B, 4A, 4B, 7B, with the strongest QTL on 2B. Eight QTL were identified for TKW, three of which located on chromosome 3A, two on 1B and one on 4B, 5A, and 5B. Only small overlapping was found among QTL for GYS, TKW, and GPC, and increasing alleles coming from both parents on different chromosomes. Good candidate genes were identified in the QTL confidence intervals for GYS and TKW.&amp;lt;/p&amp;gt;;,citation_author=Angelica Giancaspro;,citation_author=Stefania L. Giove;,citation_author=Antonio Blanco;,citation_author=Agata Gadaleta;,citation_publication_date=2019-11;,citation_cover_date=2019-11;,citation_year=2019;,citation_fulltext_html_url=https://www.frontiersin.org/journals/plant-science/articles/10.3389/fpls.2019.01509/full;,citation_doi=10.3389/fpls.2019.01509;,citation_issn=1664-462X;,citation_volume=10;,citation_journal_title=Frontiers in Plant Science;">
<meta name="citation_reference" content="citation_title=Industrial Hemp and Hemp Byproducts as Sustainable Feedstuffs in Livestock Diets;,citation_abstract=Cannabis sativa, more commonly referred to as industrial hemp, has been used for fiber, food, and medicinal purposes for millennia. Temporarily suffering a setback for parts of the 20th and 21st centuries, interest in hemp production has resurged with the recent rollback of legislative restrictions. Most notably, an interest in the nutritional profile of hemp seed has developed based on its rich array of fatty acids and protein. The increase in hemp production has led to a reciprocating increase in the availability of hemp seed and its associated by-products. While anecdotal and historical accounts exist for feeding hemp to livestock, published reports are limited. This is further complicated by a lack of acceptance in many countries for feeding hemp to livestock. Livestock’s ability to consume unusable waste products from various commodities and turn them into consumable goods is a major component of sustainable agriculture. Research indicates that dietary enrichment with hemp has the potential as a valuable source of protein, fat, and fiber to increase the quality and composition of meat, milk, and eggs; however, to what extent remains unknown. This review aims to explore the nutrient profile of the industrial hemp plant and evaluate our current understanding of hemp used in livestock diets.;,citation_author=Kristine Ely;,citation_author=John Fike;,citation_editor=Dinesh Chandra Agrawal;,citation_editor=Rajiv Kumar;,citation_editor=Muralikrishnan Dhanasekaran;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://doi.org/10.1007/978-981-16-8778-5_6;,citation_doi=10.1007/978-981-16-8778-5_6;,citation_isbn=9789811687785;,citation_inbook_title=Cannabis/Hemp for Sustainable Agriculture and Materials;">
<meta name="citation_reference" content="citation_title=Heritability and Associations among Grain Yield and Quality Traits in Quality Protein Maize (QPM) and Non-QPM Hybrids;,citation_abstract=Maize (Zea mays L.) is the main staple cereal food crop cultivated in southern Africa. Interactions between grain yield and biochemical traits can be useful to plant breeders in making informed decisions on the traits to be considered in breeding programs for high grain yield and enhanced quality. The objectives of this study were to estimate the heritability of grain yield and its related traits, as well as quality traits, and determine the association between quality protein maize (QPM) with non-QPM crosses. Grain yield, and agronomic and quality trait data were obtained from 13 field trials in two countries, for two consecutive seasons. Significant genotypic and phenotypic correlations were recorded for grain yield with protein content (rG = 0.38; rP = 0.25), and tryptophan with oil content (rG = 0.58; rP = 0.25), and negative rG and rP correlations were found for protein with tryptophan content and grain yield with tryptophan content. Path analysis identified ear aspect, ears per plant, and starch as the major traits contributing to grain yield. It is recommended that ear aspect should be considered a key secondary trait in breeding for QPM hybrids. The negative association between grain yield and tryptophan, and between protein and tryptophan, will make it difficult to develop hybrids with high grain yield and high tryptophan content. Hence, it is recommended that gene pyramiding should be considered for these traits.;,citation_author=Isaac Kodzo Amegbor;,citation_author=Angeline Biljon;,citation_author=Nemera Shargie;,citation_author=Amsal Tarekegne;,citation_author=Maryke T. Labuschagne;,citation_publication_date=2022-03;,citation_cover_date=2022-03;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8951685/;,citation_issue=6;,citation_doi=10.3390/plants11060713;,citation_issn=2223-7747;,citation_pmid=35336595;,citation_volume=11;,citation_journal_title=Plants;">
<meta name="citation_reference" content="citation_title=Direct Spectrophotometric Determination of Fat and Moisture in Meat Products;,citation_abstract=SUMMARY— The near-infrared spectral absorption properties of P-mm-thick samples of meat emulsions were measured by direct spectrophotometric techniques. The resulting spectra are interpreted in terms of absorptions from O-H and C-H stretching vibrations combined with scatter losses. Optical-density differences are correlated with fat and moisture contents. The difference in optical density between 1.80 and 1.725 p gave a high correlation with moisture content and the difference between 1.725 and 1.65 P gave a high correlation with fat content. Direct spectrophotometric analysis predicted fat content within a standard error of ± 2.1% and moisture content within ± 1.4%. The possibilities of this technique are explored and the problems to be solved in developing a rapid, accurate method are discussed;,citation_author=Itamar Ben-Gera;,citation_author=Karl H. Norris;,citation_publication_date=1968;,citation_cover_date=1968;,citation_year=1968;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1365-2621.1968.tb00885.x;,citation_issue=1;,citation_doi=10.1111/j.1365-2621.1968.tb00885.x;,citation_issn=1750-3841;,citation_volume=33;,citation_journal_title=Journal of Food Science;">
<meta name="citation_reference" content="citation_title=Measuring Protein Content in Food: An Overview of Methods;,citation_abstract=In order to determine the quantity of protein in food, it is important to have standardized analytical methods. Several methods exist that are used in different food industries to quantify protein content, including the Kjeldahl, Lowry, Bradford and total amino acid content methods. The correct determination of the protein content of foods is important as, often, as is the case with milk, it determines the economic value of the food product and it can impact the economic feasibility of new industries for alternative protein production. This editorial provides an overview of different protein determination methods and describes their advantages and disadvantages.;,citation_author=Maria Hayes;,citation_publication_date=2020-09;,citation_cover_date=2020-09;,citation_year=2020;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7597951/;,citation_issue=10;,citation_doi=10.3390/foods9101340;,citation_issn=2304-8158;,citation_pmid=32977393;,citation_volume=9;,citation_journal_title=Foods;">
<meta name="citation_reference" content="citation_title=Building Predictive Models in R Using the caret Package;,citation_author=Kuhn;,citation_author=Max;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_fulltext_html_url=https://www.jstatsoft.org/index.php/jss/article/view/v028i05;,citation_issue=5;,citation_doi=10.18637/jss.v028.i05;,citation_volume=28;,citation_journal_title=Journal of Statistical Software;">
<meta name="citation_reference" content="citation_title=Pls: Partial Least Squares and Principal Component Regression;,citation_author=Kristian Hovde Liland;,citation_author=Bjørn-Helge Mevik;,citation_author=Ron Wehrens;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=pls;">
<meta name="citation_reference" content="citation_title=Welcome to the tidyverse;,citation_author=Hadley Wickham;,citation_author=Mara Averick;,citation_author=Jennifer Bryan;,citation_author=Winston Chang;,citation_author=Lucy D’Agostino McGowan;,citation_author=Romain François;,citation_author=Garrett Grolemund;,citation_author=Alex Hayes;,citation_author=Lionel Henry;,citation_author=Jim Hester;,citation_author=Max Kuhn;,citation_author=Thomas Lin Pedersen;,citation_author=Evan Miller;,citation_author=Stephan Milton Bache;,citation_author=Kirill Müller;,citation_author=Jeroen Ooms;,citation_author=David Robinson;,citation_author=Dana Paige Seidel;,citation_author=Vitalie Spinu;,citation_author=Kohske Takahashi;,citation_author=Davis Vaughan;,citation_author=Claus Wilke;,citation_author=Kara Woo;,citation_author=Hiroaki Yutani;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=43;,citation_doi=10.21105/joss.01686;,citation_volume=4;,citation_journal_title=Journal of Open Source Software;">
<meta name="citation_reference" content="citation_title=R: A Language and Environment for Statistical Computing;,citation_author=R Core Team;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://www.R-project.org/;">
<meta name="citation_reference" content="citation_title=Data.table: Extension of ‘data.frame‘;,citation_author=Tyson Barrett;,citation_author=Matt Dowle;,citation_author=Arun Srinivasan;,citation_author=Jan Gorecki;,citation_author=Michael Chirico;,citation_author=Toby Hocking;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=data.table;">
<meta name="citation_reference" content="citation_title=Proteomic characterization of hempseed (&amp;amp;amp;lt;i&amp;gt;Cannabis sativa&amp;lt;/i&amp;gt; L.);,citation_abstract=This paper presents an investigation on hempseed proteome. The experimental approach, based on combinatorial peptide ligand libraries (CPLLs), SDS-PAGE separation, nLC-ESI-MS/MS identification, and database search, permitted identifying in total 181 expressed proteins. This very large number of identifications was achieved by searching in two databases: Cannabis sativa L. (56 gene products identified) and Arabidopsis thaliana (125 gene products identified). By performing a protein–protein association network analysis using the STRING software, it was possible to build the first interactomic map of all detected proteins, characterized by 137 nodes and 410 interactions. Finally, a Gene Ontology analysis of the identified species permitted to classify their molecular functions: the great majority is involved in the seed metabolic processes (41%), responses to stimulus (8%), and biological process (7%). Biological significance Hempseed is an underexploited non-legume protein-rich seed. Although its protein is well known for its digestibility, essential amino acid composition, and useful techno-functional properties, a comprehensive proteome characterization is still lacking. The objective of this work was to fill this knowledge gap and provide information useful for a better exploitation of this seed in different food products.;,citation_author=Gilda Aiello;,citation_author=Elisa Fasoli;,citation_author=Giovanna Boschin;,citation_author=Carmen Lammi;,citation_author=Chiara Zanoni;,citation_author=Attilio Citterio;,citation_author=Anna Arnoldi;,citation_publication_date=2016-09;,citation_cover_date=2016-09;,citation_year=2016;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S1874391916302354;,citation_doi=10.1016/j.jprot.2016.05.033;,citation_issn=1874-3919;,citation_volume=147;,citation_journal_title=Journal of Proteomics;,citation_series_title=Foodomics - Novel insights in food and nutrition domains;">
<meta name="citation_reference" content="citation_title=Sorghum Grains Grading for Food, Feed, and Fuel Using NIR Spectroscopy;,citation_abstract=&amp;amp;amp;lt;p&amp;gt;Near-infrared spectroscopy (NIR) is a non-destructive, fast, and low-cost method to measure the grain quality of different cereals. However, the feasibility for determining the critical biochemicals, related to the classifications for food, feed, and fuel products are not adequately investigated. Fourier-transform (FT) NIR was applied in this study to determine the eight biochemicals in four types of sorghum samples: hulled grain flours, hull-less grain flours, whole grains, and grain flours. A total of 20 hybrids of sorghum grains were selected from the two locations in China. Followed by FT-NIR spectral and wet-chemically measured biochemical data, partial least squares regression (PLSR) was used to construct the prediction models. The results showed that sorghum grain morphology and sample format affected the prediction of biochemicals. Using NIR data of grain flours generally improved the prediction compared with the use of NIR data of whole grains. In addition, using the spectra of whole grains enabled comparable predictions, which are recommended when a non-destructive and rapid analysis is required. Compared with the hulled grain flours, hull-less grain flours allowed for improved predictions for tannin, cellulose, and hemicellulose using NIR data. This study aimed to provide a reference for the evaluation of sorghum grain biochemicals for food, feed, and fuel without destruction and complex chemical analysis.&amp;lt;/p&amp;gt;;,citation_author=Irsa Ejaz;,citation_author=Siyang He;,citation_author=Wei Li;,citation_author=Naiyue Hu;,citation_author=Chaochen Tang;,citation_author=Songbo Li;,citation_author=Meng Li;,citation_author=Boubacar Diallo;,citation_author=Guanghui Xie;,citation_author=Kang Yu;,citation_publication_date=2021-09;,citation_cover_date=2021-09;,citation_year=2021;,citation_fulltext_html_url=https://www.frontiersin.org/journals/plant-science/articles/10.3389/fpls.2021.720022/full;,citation_doi=10.3389/fpls.2021.720022;,citation_issn=1664-462X;,citation_volume=12;,citation_journal_title=Frontiers in Plant Science;">
<meta name="citation_reference" content="citation_title=Maturity, Variety and Origin Determination in White Grapes (Vitis Vinifera L.) Using near Infrared Reflectance Technology;,citation_fulltext_html_url=https://journals.sagepub.com/doi/epdf/10.1255/jnirs.566;,citation_doi=10.1255/jnirs.566;">
<meta name="citation_reference" content="citation_title=Critical review of chemometric indicators commonly used for assessing the quality of the prediction of soil attributes by NIR spectroscopy;,citation_abstract=Near-infrared (NIR) and mid-IR spectroscopy applied to soil compositional analysis started to develop markedly in the 1990s, taking advantage of earlier advances in instrumentation and chemometrics for agricultural products. Today, NIR spectroscopy is envisioned as replacing laboratory analysis in certain applications (e.g., soil-carbon-credit assessment at the farm level). However, accuracy is still unsatisfactory compared with standard laboratory procedures, leading some authors to think that such a challenge will never be met. This article investigates the critical points to be aware of when accuracy of NIR-based measurements is assessed. First is the decomposition of the standard error of prediction into components of bias and variance, only the latter being reducible by averaging. This decomposition is not used routinely in the soil-science literature. Contrarily, a log-normal distribution of reference values is very often encountered with soil samples [e.g., elemental concentrations (e.g., carbon)] with numerous small or zero values. These very skewed distributions make us take precautions when using inverse regression methods (e.g., principal component regression or partial least squares), which force the predictions towards the centre of the calibration set, leading to negative effects on the standard error prediction – and therefore on prediction accuracy – especially when log-normal distributions are encountered. Such distributions, which are very common for soil components, also make the ratio of performance to deviation a useless, even hazardous, tool, leading to erroneous conclusions. We propose a new index based on the quartiles of the empirical distribution – ratio of performance to inter-quartile distance – to overcome this problem.;,citation_author=Véronique Bellon-Maurel;,citation_author=Elvira Fernandez-Ahumada;,citation_author=Bernard Palagos;,citation_author=Jean-Michel Roger;,citation_author=Alex McBratney;,citation_publication_date=2010-10;,citation_cover_date=2010-10;,citation_year=2010;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0165993610001585;,citation_issue=9;,citation_doi=10.1016/j.trac.2010.05.006;,citation_issn=0165-9936;,citation_volume=29;,citation_journal_title=TrAC Trends in Analytical Chemistry;">
<meta name="citation_reference" content="citation_title=Critical review of chemometric indicators commonly used for assessing the quality of the prediction of soil attributes by NIR spectroscopy;,citation_abstract=Near-infrared (NIR) and mid-IR spectroscopy applied to soil compositional analysis started to develop markedly in the 1990s, taking advantage of earlier advances in instrumentation and chemometrics for agricultural products. Today, NIR spectroscopy is envisioned as replacing laboratory analysis in certain applications (e.g., soil-carbon-credit assessment at the farm level). However, accuracy is still unsatisfactory compared with standard laboratory procedures, leading some authors to think that such a challenge will never be met. This article investigates the critical points to be aware of when accuracy of NIR-based measurements is assessed. First is the decomposition of the standard error of prediction into components of bias and variance, only the latter being reducible by averaging. This decomposition is not used routinely in the soil-science literature. Contrarily, a log-normal distribution of reference values is very often encountered with soil samples [e.g., elemental concentrations (e.g., carbon)] with numerous small or zero values. These very skewed distributions make us take precautions when using inverse regression methods (e.g., principal component regression or partial least squares), which force the predictions towards the centre of the calibration set, leading to negative effects on the standard error prediction – and therefore on prediction accuracy – especially when log-normal distributions are encountered. Such distributions, which are very common for soil components, also make the ratio of performance to deviation a useless, even hazardous, tool, leading to erroneous conclusions. We propose a new index based on the quartiles of the empirical distribution – ratio of performance to inter-quartile distance – to overcome this problem.;,citation_author=Véronique Bellon-Maurel;,citation_author=Elvira Fernandez-Ahumada;,citation_author=Bernard Palagos;,citation_author=Jean-Michel Roger;,citation_author=Alex McBratney;,citation_publication_date=2010-10;,citation_cover_date=2010-10;,citation_year=2010;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0165993610001585;,citation_issue=9;,citation_doi=10.1016/j.trac.2010.05.006;,citation_issn=0165-9936;,citation_volume=29;,citation_journal_title=TrAC Trends in Analytical Chemistry;">
<meta name="citation_reference" content="citation_title=Dual-sPLS: A family of Dual Sparse Partial Least Squares regressions for feature selection and prediction with tunable sparsity; evaluation on simulated and near-infrared (NIR) data;,citation_abstract=Relating a set of variables X to a response y is crucial in chemometrics. A quantitative prediction objective can be enriched by qualitative data interpretation, for instance by locating the most influential features. When high-dimensional problems arise, dimension reduction techniques can be used. Most notable are projections (e.g. Partial Least Squares or PLS ) or variable selections (e.g. lasso). Sparse partial least squares combine both strategies, by blending variable selection into PLS. The variant presented in this paper, Dual-sPLS, generalizes the classical PLS1 algorithm. It provides balance between accurate prediction and efficient interpretation. It is based on penalizations inspired by classical regression methods (lasso, group lasso, least squares, ridge) and uses the dual norm notion. The resulting sparsity is enforced by an intuitive shrinking ratio parameter. Dual-sPLS favorably compares to similar regression methods, on simulated and real chemical data.;,citation_author=Louna Alsouki;,citation_author=Laurent Duval;,citation_author=Clément Marteau;,citation_author=Rami El Haddad;,citation_author=François Wahl;,citation_publication_date=2023-06;,citation_cover_date=2023-06;,citation_year=2023;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0169743923000631;,citation_doi=10.1016/j.chemolab.2023.104813;,citation_issn=0169-7439;,citation_volume=237;,citation_journal_title=Chemometrics and Intelligent Laboratory Systems;">
<meta name="citation_reference" content="citation_title=PLS-regression: A basic tool of chemometrics;,citation_abstract=PLS-regression (PLSR) is the PLS approach in its simplest, and in chemistry and technology, most used form (two-block predictive PLS). PLSR is a method for relating two data matrices, X and Y, by a linear multivariate model, but goes beyond traditional regression in that it models also the structure of X and Y. PLSR derives its usefulness from its ability to analyze data with many, noisy, collinear, and even incomplete variables in both X and Y. PLSR has the desirable property that the precision of the model parameters improves with the increasing number of relevant variables and observations. This article reviews PLSR as it has developed to become a standard tool in chemometrics and used in chemistry and engineering. The underlying model and its assumptions are discussed, and commonly used diagnostics are reviewed together with the interpretation of resulting parameters. Two examples are used as illustrations: First, a Quantitative Structure–Activity Relationship (QSAR)/Quantitative Structure–Property Relationship (QSPR) data set of peptides is used to outline how to develop, interpret and refine a PLSR model. Second, a data set from the manufacturing of recycled paper is analyzed to illustrate time series modelling of process data by means of PLSR and time-lagged X-variables.;,citation_author=Svante Wold;,citation_author=Michael Sjöström;,citation_author=Lennart Eriksson;,citation_publication_date=2001-10;,citation_cover_date=2001-10;,citation_year=2001;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0169743901001551;,citation_issue=2;,citation_doi=10.1016/S0169-7439(01)00155-1;,citation_issn=0169-7439;,citation_volume=58;,citation_journal_title=Chemometrics and Intelligent Laboratory Systems;,citation_series_title=PLS Methods;">
<meta name="citation_reference" content="citation_title=Feasibility study of NIR for surveillance of a pharmaceutical process, including a study of different preprocessing techniques;,citation_abstract=Multivariate calibration models for predicting the initial reactant and an impurity in a synthesis of contrast media have been developed using near-infrared (NIR) spectroscopy. The reaction is presently monitored off-line by HPLC. The calibration models are based on spectra measured on-line. The process solution is chemically complex and the intention of this study was to enhance knowledge about the predictive ability of the models. Different data-preprocessing and variable selection methods reported in the literature have been tested in order to enhance chemical information and reduce irrelevant variability due to physical influences on measured spectra such as scattering and temperature. In addition to standard preprocessing methods such as normalization, differentiation and multiplicative scatter correction, orthogonal signal correction and optimized scaling were tested. The modelling was performed by partial least squares (PLS), except for optimized scaling which uses principal component regression (PCR). Optimized scaling on differentiated data gave the lowest prediction error. The NIR region from 1100 to 1900 nm gave better models than variable selection and fewer variables. Models with correlation coefficients for best fit equal to 0.998 and 0.961 were obtained for the initial reactant and the impurity, respectively. Copyright © 2002 John Wiley &amp;amp;amp; Sons, Ltd.;,citation_author=Laila Stordrange;,citation_author=Fred Olav Libnau;,citation_author=Dick Malthe-Sørenssen;,citation_author=Olav M. Kvalheim;,citation_publication_date=2002;,citation_cover_date=2002;,citation_year=2002;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1002/cem.754;,citation_issue=8-10;,citation_doi=10.1002/cem.754;,citation_issn=1099-128X;,citation_volume=16;,citation_journal_title=Journal of Chemometrics;">
<meta name="citation_reference" content="citation_title=Partial Least Squares (PLS) Regression.;,citation_author=Herve Abdi;">
<meta name="citation_reference" content="citation_title=Standard Normal Variate Transformation and De-Trending of Near-Infrared Diffuse Reflectance Spectra;,citation_abstract=Particle size, scatter, and multi-collinearity are long-standing problems encountered in diffuse reflectance spectrometry. Multiplicative combinations of these effects are the major factor inhibiting the interpretation of near-infrared diffuse reflectance spectra. Sample particle size accounts for the majority of the variance, while variance due to chemical composition is small. Procedures are presented whereby physical and chemical variance can be separated. Mathematical transformations—standard normal variate (SNV) and de-trending (DT)—applicable to individual NIR diffuse reflectance spectra are presented. The standard normal variate approach effectively removes the multiplicative interferences of scatter and particle size. De-trending accounts for the variation in baseline shift and curvilinearity, generally found in the reflectance spectra of powdered or densely packed samples, with the use of a second-degree polynomial regression. NIR diffuse reflectance spectra transposed by these methods are free from multi-collinearity and are not confused by the complexity of shape encountered with the use of derivative spectroscopy.;,citation_author=R. J. Barnes;,citation_author=M. S. Dhanoa;,citation_author=Susan J. Lister;,citation_publication_date=1989;,citation_cover_date=1989;,citation_year=1989;,citation_fulltext_html_url=https://doi.org/10.1366/0003702894202201;,citation_issue=5;,citation_doi=10.1366/0003702894202201;,citation_volume=43;,citation_journal_title=Applied Spectroscopy;">
<meta name="citation_reference" content="citation_title=Comparison of grating-based near-infrared (NIR) and Fourier transform mid-infrared (ATR-FT/MIR) spectroscopy based on spectral preprocessing and wavelength selection for the determination of crude protein and moisture content in wheat;,citation_abstract=Growing trends towards the development of rapid, simple, and cost-effective methodologies for grain quality monitoring have been observed for a long time. Infrared spectroscopy associated with chemometrics has proven to be a promising alternative to traditional wet chemistry methods. So far comparative research that focuses on the relation between spectral preprocessing, wavelength selection, and model performance is rare. This paper aims to explore the effect of spectral preprocessing and wavelengths selection on the performance of partial least square (PLS) regression models constructed with near-infrared (NIR) and mid-infrared (ATR-FT/MIR) spectroscopy. Eighty wheat samples were randomly selected in Western Canada. NIR (680–2500&nbsp;nm) and MIR (4000–700&nbsp;cm−1) spectra were collected and calibrated with reference crude protein (CP) and moisture content using PLS regression technique based on different spectral preprocessing methods and selected important wavelengths. In a general way, the models generated with NIR spectra showed superior performance than that developed with MIR spectra. For CP, the best NIR model was developed with the 1400–2500&nbsp;nm spectral range of standard normal variate (SNV) and first derivative (FD) pretreated spectra, which showed an excellent prediction performance (R2&nbsp;=&nbsp;0.97); the best MIR model was established using the 1750–1100&nbsp;cm−1 region of SNV corrected spectra, which gave a good predictive ability (R2&nbsp;=&nbsp;0.90). Regarding to the moisture, the best model obtained by NIR technique (based on the 1100–2500&nbsp;nm region of raw spectra) showed a good prediction performance (R2&nbsp;=&nbsp;0.86), while the best model generated by MIR technique (using the full wavenumber range of FD-SNV preprocessed spectra) only gave approximate quantitative prediction (R2&nbsp;=&nbsp;0.72). Both spectral preprocessing and wavelength selection could affect the performance of PLS models for predicting CP and moisture content in wheat. More effort is needed to improve the performance of ATR-FT/MIR spectroscopy for quantitative analysis based on appropriate chemometrics tools.;,citation_author=Haitao Shi;,citation_author=Peiqiang Yu;,citation_publication_date=2017-12;,citation_cover_date=2017-12;,citation_year=2017;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0956713517303134;,citation_doi=10.1016/j.foodcont.2017.06.015;,citation_issn=0956-7135;,citation_volume=82;,citation_journal_title=Food Control;">
<meta name="citation_reference" content="citation_title=Quantitative analysis of honey adulteration by spectrum analysis combined with several high-level data fusion strategies;,citation_abstract=Illegal addition of syrup to honey products would refdeduce the natural nutritional value of honey. It is essential to develop convenient and accurate analytical methods to rapidly monitor products quality. This study investigated the potential usability of combining mid infrared(MIR) and Raman for the quantification of HFGS (high fructose syrup) in bass honey blends. Where in the partial least squares (PLS) model was established to predict the concentration of the adulterant. Three data fusion strategies (low, mid and high-level) were applied to take advantage of the synergistic effect of chemical information from MIR and Raman, and adopted three high-level data fusion strategies to improve the prediction accuracy of the quantitative model. After data fusion, the RMSEP of high-level (1–3) were better than MIR (1.93) and Raman (2.30). Besides, calibration coefficient of determination (R2) and the RMSECV of high-level data fusion (1–3)also outperformed Raman (0.80, 3.38) and MIR (0.95,1.63).Results showed that: (1) The chemical information obtained from MIR and Raman have complementary advantages. (2) The three high-level fusion strategies can be used as a reliable tool for quantitative analysis.;,citation_author=Yang Li;,citation_author=Yue Huang;,citation_author=Jingjing Xia;,citation_author=Yanmei Xiong;,citation_author=Shungeng Min;,citation_publication_date=2020-05;,citation_cover_date=2020-05;,citation_year=2020;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0924203120300631;,citation_doi=10.1016/j.vibspec.2020.103060;,citation_issn=0924-2031;,citation_volume=108;,citation_journal_title=Vibrational Spectroscopy;">
<meta name="citation_reference" content="citation_title=Properties of Savitzky–Golay digital differentiators;,citation_abstract=The Savitzky–Golay (SG) filters are generally used for smoothing and differentiation in many fields. The properties of the SG smoothing filters have been well studied. However, the properties of the SG differentiation filters or SG digital differentiators (SGDD, for the first order differentiation) are not developed well somehow, although they have been widely used. In the paper, the properties of the SGDD are discussed in detail. The effects of the SGDD on a noise-free single Gaussian line and a noise-free Gaussian doublet are studied via simulation. The results indicate that the contrast and resolution loss of the SGDD depends on the ratio of the width of signal derivative and filter length. The linear SGDD is not preferable for preserving small details of signal derivative from this study. The cubic and quintic SGDD with the choice of appropriate filter length are recommended in order to maintain the resolution of signal derivative.;,citation_author=Jianwen Luo;,citation_author=Kui Ying;,citation_author=Ping He;,citation_author=Jing Bai;,citation_publication_date=2005-03;,citation_cover_date=2005-03;,citation_year=2005;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S1051200404000727;,citation_issue=2;,citation_doi=10.1016/j.dsp.2004.09.008;,citation_issn=1051-2004;,citation_volume=15;,citation_journal_title=Digital Signal Processing;">
<meta name="citation_reference" content="citation_title=Back to basics: Spectral pre-treatments, derivatives;,citation_abstract=This column is about the most basic of pre-treatments, which has been used in spectroscopy well before the word &amp;amp;amp;quot;Chemometrics&amp;quot; was invented.;,citation_author=Spectroscopy Europe;,citation_fulltext_html_url=https://www.spectroscopyeurope.com/td-column/back-basics-spectral-pre-treatments-derivatives;">
<meta name="citation_reference" content="citation_title=Understanding and Correcting the Factors Which Affect Diffuse Transmittance Spectra;,citation_fulltext_html_url=https://journals.sagepub.com/doi/epdf/10.1255/nirn.613;,citation_doi=10.1255/nirn.613;">
<meta name="citation_reference" content="citation_title=Review of the most common pre-processing techniques for near-infrared spectra;,citation_abstract=Pre-processing of near-infrared (NIR) spectral data has become an integral part of chemometrics modeling. The objective of the pre-processing is to remove physical phenomena in the spectra in order to improve the subsequent multivariate regression, classification model or exploratory analysis. The most widely used pre-processing techniques can be divided into two categories: scatter-correction methods and spectral derivatives. This review describes and compares the theoretical and algorithmic foundations of current pre-processing methods plus the qualitative and quantitative consequences of their application. The aim is to provide NIR users with better end-models through fundamental knowledge on spectral pre-processing.;,citation_author=Åsmund Rinnan;,citation_author=Frans Berg;,citation_author=Søren Balling Engelsen;,citation_publication_date=2009-11;,citation_cover_date=2009-11;,citation_year=2009;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0165993609001629;,citation_issue=10;,citation_doi=10.1016/j.trac.2009.07.007;,citation_issn=0165-9936;,citation_volume=28;,citation_journal_title=TrAC Trends in Analytical Chemistry;">
<meta name="citation_reference" content="citation_title=Visible and near-infrared spectroscopy predicted leaf nitrogen contents of potato varieties under different growth and management conditions;,citation_abstract=Visible-Near Infrared (vis-NIR) spectroscopy can provide a faster, cost-effective, and user-friendly solution to monitor leaf N status, potentially overcoming the limitations of current techniques. The objectives of the study were to develop and validate partial least square regression (PLSR) to estimate the total N contents of fresh and removed leaves of potatoes using the vis-NIR spectral range (350–2500&nbsp;nm)&nbsp;generated from a handheld proximal sensor. The model was built using data collected from Hancock Agricultural Research Station, WI, USA in 2020 and was validated using samples collected in 2021 for four different conditions. The conditions included two sites (Coloma and Hancock), four potato varieties (Burbank, Norkotah, Goldrush, and Silverton), two N rates (unfertilized and 308&nbsp;kg N ha−1), and four growth stages (vegetative, tuber initiation, tuber bulking, and tuber maturation). The calibration and validation models had high predictive performance for leaf total N with R2 &amp;amp;amp;gt; 0.8 and RPD &amp;gt; 2. The model accuracy was affected by the total N contents in the leaf samples where the model underpredicted the samples with total leaf N contents greater than 6%.;,citation_author=Ashmita Rawal;,citation_author=Alfred Hartemink;,citation_author=Yakun Zhang;,citation_author=Yi Wang;,citation_author=Richard A. Lankau;,citation_author=Matthew D. Ruark;,citation_publication_date=2024-04;,citation_cover_date=2024-04;,citation_year=2024;,citation_fulltext_html_url=https://doi.org/10.1007/s11119-023-10091-z;,citation_issue=2;,citation_doi=10.1007/s11119-023-10091-z;,citation_issn=1573-1618;,citation_volume=25;,citation_journal_title=Precision Agriculture;">
<meta name="citation_reference" content="citation_title=Prediction of total carbon, total nitrogen, and pH of organic materials using visible near-infrared reflectance spectroscopy;,citation_abstract=We used visible near-infrared reflectance spectroscopy (VNIRS) to characterize a set of organic materials (n = 75). Good predictions for total carbon, excellent predictions for total nitrogen, but poor predictions for pH were found. This preliminary study showed that VNIRS could rapidly characterize organic materials before soil application.;,citation_author=Mervin St Luce;,citation_author=Noura Ziadi;,citation_author=Bernard Gagnon;,citation_author=Vicky Lévesque;,citation_publication_date=2017-12;,citation_cover_date=2017-12;,citation_year=2017;,citation_fulltext_html_url=https://bioone.org/journals/canadian-journal-of-soil-science/volume-98/issue-1/cjss-2017-0109/Prediction-of-total-carbon-total-nitrogen-and-pH-of-organic/10.1139/cjss-2017-0109.full;,citation_issue=1;,citation_doi=10.1139/cjss-2017-0109;,citation_issn=0008-4271, 1918-1841;,citation_volume=98;,citation_journal_title=Canadian Journal of Soil Science;">
<meta name="citation_reference" content="citation_title=R: A language and environment for statistical computing;,citation_author=R Core Team;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://www.R-project.org/;">
<meta name="citation_reference" content="citation_title=Building predictive models in r using the caret package;,citation_author=Kuhn;,citation_author=Max;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_fulltext_html_url=https://www.jstatsoft.org/index.php/jss/article/view/v028i05;,citation_issue=5;,citation_doi=10.18637/jss.v028.i05;,citation_volume=28;,citation_journal_title=Journal of Statistical Software;">
<meta name="citation_reference" content="citation_title=data.table: Extension of “data.frame”;,citation_author=Tyson Barrett;,citation_author=Matt Dowle;,citation_author=Arun Srinivasan;,citation_author=Jan Gorecki;,citation_author=Michael Chirico;,citation_author=Toby Hocking;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=data.table;">
<meta name="citation_reference" content="citation_title=emmeans: Estimated marginal means, aka least-squares means;,citation_author=Russell V. Lenth;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=emmeans;">
<meta name="citation_reference" content="citation_title=nlme: Linear and nonlinear mixed effects models;,citation_author=José Pinheiro;,citation_author=Douglas Bates;,citation_author=R Core Team;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=nlme;">
<meta name="citation_reference" content="citation_title=Mixed-effects models in s and s-PLUS;,citation_author=José C. Pinheiro;,citation_author=Douglas M. Bates;,citation_publication_date=2000;,citation_cover_date=2000;,citation_year=2000;,citation_doi=10.1007/b98882;">
<meta name="citation_reference" content="citation_title=pls: Partial least squares and principal component regression;,citation_author=Kristian Hovde Liland;,citation_author=Bjørn-Helge Mevik;,citation_author=Ron Wehrens;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=pls;">
<meta name="citation_reference" content="citation_title=An introduction to the prospectr package;,citation_author=Antoine Stevens;,citation_author=Leornardo Ramirez-Lopez;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;">
<meta name="citation_reference" content="citation_title=skimr: Compact and flexible summaries of data;,citation_author=Elin Waring;,citation_author=Michael Quinn;,citation_author=Amelia McNamara;,citation_author=Eduardo Arino de la Rubia;,citation_author=Hao Zhu;,citation_author=Shannon Ellis;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://CRAN.R-project.org/package=skimr;">
<meta name="citation_reference" content="citation_title=Tidymodels: A collection of packages for modeling and machine learning using tidyverse principles.;,citation_author=Max Kuhn;,citation_author=Hadley Wickham;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://www.tidymodels.org;">
<meta name="citation_reference" content="citation_title=Welcome to the tidyverse;,citation_author=Hadley Wickham;,citation_author=Mara Averick;,citation_author=Jennifer Bryan;,citation_author=Winston Chang;,citation_author=Lucy D’Agostino McGowan;,citation_author=Romain François;,citation_author=Garrett Grolemund;,citation_author=Alex Hayes;,citation_author=Lionel Henry;,citation_author=Jim Hester;,citation_author=Max Kuhn;,citation_author=Thomas Lin Pedersen;,citation_author=Evan Miller;,citation_author=Stephan Milton Bache;,citation_author=Kirill Müller;,citation_author=Jeroen Ooms;,citation_author=David Robinson;,citation_author=Dana Paige Seidel;,citation_author=Vitalie Spinu;,citation_author=Kohske Takahashi;,citation_author=Davis Vaughan;,citation_author=Claus Wilke;,citation_author=Kara Woo;,citation_author=Hiroaki Yutani;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=43;,citation_doi=10.21105/joss.01686;,citation_volume=4;,citation_journal_title=Journal of Open Source Software;">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Near Infrared Spectroscopy Predicts Percentage Crude Protein in Hemp Grain</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
            <div class="quarto-title-meta-author">
          <div class="quarto-title-meta-heading">Authors</div>
          <div class="quarto-title-meta-heading">Affiliations</div>
          
                <div class="quarto-title-meta-contents">
            <p class="author">Ryan V. Crawford <a href="mailto:rvc3@cornell.edu" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0009-0006-3052-3269" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Cornell University
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Jamie Crawford </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Cornell University
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Julie L. Hansen <a href="https://orcid.org/0000-0001-7247-9186" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Cornell University
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Lawrence B. Smart <a href="https://orcid.org/0000-0002-7812-7736" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Cornell AgriTech
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Virginia Moore <a href="https://orcid.org/0000-0001-7888-3366" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Cornell University
                      </p>
                  </div>
                    </div>
        
        <div class="quarto-title-meta">

                      
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">April 3, 2024</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      <div class="quarto-alternate-formats"><div class="quarto-title-meta-heading">Other Formats</div><div class="quarto-title-meta-contents"><p><a href="index.docx"><i class="bi bi-file-word"></i>MS Word</a></p></div><div class="quarto-title-meta-contents"><p><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF (agu)</a></p></div><div class="quarto-title-meta-contents"><p><a href="index-meca.zip" data-meca-link="true"><i class="bi bi-archive"></i>MECA Bundle</a></p></div></div></div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Hemp, Grain, Spectroscopy</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> INTRODUCTION</a></li>
  <li><a href="#materials-and-methods" id="toc-materials-and-methods" class="nav-link" data-scroll-target="#materials-and-methods"><span class="header-section-number">2</span> MATERIALS AND METHODS</a>
  <ul class="collapse">
  <li><a href="#hemp-grain-sample-background" id="toc-hemp-grain-sample-background" class="nav-link" data-scroll-target="#hemp-grain-sample-background"><span class="header-section-number">2.1</span> Hemp Grain Sample Background</a></li>
  <li><a href="#spectral-data-collection-and-preprocessing" id="toc-spectral-data-collection-and-preprocessing" class="nav-link" data-scroll-target="#spectral-data-collection-and-preprocessing"><span class="header-section-number">2.2</span> Spectral Data Collection and Preprocessing</a></li>
  <li><a href="#laboratory-validation" id="toc-laboratory-validation" class="nav-link" data-scroll-target="#laboratory-validation"><span class="header-section-number">2.3</span> Laboratory Validation</a></li>
  <li><a href="#model-development" id="toc-model-development" class="nav-link" data-scroll-target="#model-development"><span class="header-section-number">2.4</span> Model Development</a></li>
  <li><a href="#additional-software-used" id="toc-additional-software-used" class="nav-link" data-scroll-target="#additional-software-used"><span class="header-section-number">2.5</span> Additional software used</a></li>
  </ul></li>
  <li><a href="#results-and-discussion" id="toc-results-and-discussion" class="nav-link" data-scroll-target="#results-and-discussion"><span class="header-section-number">3</span> RESULTS AND DISCUSSION</a>
  <ul class="collapse">
  <li><a href="#laboratory-assay-cp-values" id="toc-laboratory-assay-cp-values" class="nav-link" data-scroll-target="#laboratory-assay-cp-values"><span class="header-section-number">3.1</span> Laboratory assay %CP values</a></li>
  <li><a href="#preprocessing-methods-comparison" id="toc-preprocessing-methods-comparison" class="nav-link" data-scroll-target="#preprocessing-methods-comparison"><span class="header-section-number">3.2</span> Preprocessing methods comparison</a></li>
  <li><a href="#final-model-development-and-summary" id="toc-final-model-development-and-summary" class="nav-link" data-scroll-target="#final-model-development-and-summary"><span class="header-section-number">3.3</span> Final model development and summary</a></li>
  </ul></li>
  <li><a href="#acknowledgments" id="toc-acknowledgments" class="nav-link" data-scroll-target="#acknowledgments"><span class="header-section-number">4</span> ACKNOWLEDGMENTS</a></li>
  <li><a href="#conflict-of-interest" id="toc-conflict-of-interest" class="nav-link" data-scroll-target="#conflict-of-interest"><span class="header-section-number">5</span> CONFLICT OF INTEREST</a></li>
  <li><a href="#orcid" id="toc-orcid" class="nav-link" data-scroll-target="#orcid"><span class="header-section-number">6</span> ORCID</a></li>
  <li><a href="#supplemental-material" id="toc-supplemental-material" class="nav-link" data-scroll-target="#supplemental-material"><span class="header-section-number">7</span> SUPPLEMENTAL MATERIAL</a></li>
  <li><a href="#optional-sections" id="toc-optional-sections" class="nav-link" data-scroll-target="#optional-sections"><span class="header-section-number">8</span> OPTIONAL SECTIONS</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="header-section-number">9</span> REFERENCES</a></li>
  <li><a href="#figures-and-tables" id="toc-figures-and-tables" class="nav-link" data-scroll-target="#figures-and-tables"><span class="header-section-number">10</span> FIGURES AND TABLES</a></li>
  </ul>
<div class="quarto-alternate-notebooks"><h2>Notebooks</h2><ul><li><a href="index-preview.html"><i class="bi bi-journal-code"></i>Article Notebook</a></li><li><a href="notebooks/explore-earthquakes-preview.html"><i class="bi bi-journal-code"></i>Explore Earthquakes</a></li></ul></div></nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">



  


<p><strong>first draft</strong></p>
<section id="introduction" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">1</span> INTRODUCTION</h2>
<p>Hemp (Cannabis sativa L.) is an annual crop with potential uses as a source of food or feed from grain, and bast fiber or hurd from the stalk. Hemp cultivars are commonly grown for one or both purposes and a cultivar may be referred to as a grain, fiber, or dual-purpose type. Because of protein’s nutritional importance, the protein content of a grain crop is an prime consideration for researchers, producers, and consumers. Whole hemp grain typically contains approximately 20-30% protein <span class="citation" data-cites="ely_industrial_2022 barta_proteomic_2024 callaway_hempseed_2004">(<a href="#ref-barta_proteomic_2024" role="doc-biblioref">Bárta et al., 2024</a>; <a href="#ref-callaway_hempseed_2004" role="doc-biblioref">Callaway, 2004</a>; <a href="#ref-ely_industrial_2022" role="doc-biblioref">Ely &amp; Fike, 2022</a>)</span>. Crude protein is often used as a proxy for the direct measurement of protein concentration and consists of the multiplication of nitrogen concentration by a conversion factor, often 6.25 <span class="citation" data-cites="hayes_measuring_2020">(<a href="#ref-hayes_measuring_2020" role="doc-biblioref">Hayes, 2020</a>)</span>. It may be expressed as a percentage (%CP).</p>
<p>Near-infrared spectroscopy (NIRS) technology is rapid, non-destructive, and cheap. It consists of the measurement of NIR radiation reflected and absorbed from a sample (the spectra) and the relation of the spectra to laboratory values for components such as moisture, protein, fat, or fiber <span class="citation" data-cites="roberts_near-infrared_2004">(<a href="#ref-roberts_near-infrared_2004" role="doc-biblioref">Roberts et al., 2004</a>)</span>. NIRS technology has been used since the 1970’s to assess forage %CP <span class="citation" data-cites="reeves_potential_2012 williams_application_1975">(<a href="#ref-reeves_potential_2012" role="doc-biblioref">Reeves, 2012</a>; <a href="#ref-williams_application_1975" role="doc-biblioref">Williams, 1975</a>)</span>. A NIRS calibration set often consists of samples from one species grown in many environments encompassing the range of expected values from the analyte or analytes <span class="citation" data-cites="chadalavada_nir_2022">(<a href="#ref-chadalavada_nir_2022" role="doc-biblioref">Chadalavada et al., 2022</a>)</span>. Partial least squares regression (PLSR) is a typical method used in the agricultural and food sciences to relate spectra to analyte <span class="citation" data-cites="roberts_near-infrared_2004">(<a href="#ref-roberts_near-infrared_2004" role="doc-biblioref">Roberts et al., 2004</a>)</span>. PLSR calculates components that maximize covariance between predictor and response variables. PLSR uses some number of components, often selected via cross-validation, in order to fit the regression model and is commonly used in spectroscopy because it tends to work well with highly-correlated, noisy spectral data <span class="citation" data-cites="wold_pls-regression_2001">(<a href="#ref-wold_pls-regression_2001" role="doc-biblioref">Wold et al., 2001</a>)</span>.</p>
<p>A NIRS-scanned sample of undamaged grain may used for other purposes besides its scan or it may planted as a seed. In wheat and corn, grain protein content has been shown to be heritable <span class="citation" data-cites="giancaspro_genetic_2019 geyer_genetics_2022">(<a href="#ref-geyer_genetics_2022" role="doc-biblioref">Geyer et al., 2022</a>; <a href="#ref-giancaspro_genetic_2019" role="doc-biblioref">Giancaspro et al., 2019</a>)</span>. This suggests (at least potentially) that NIRS technology could serve as resource to rapidly identify high %CP hemp germplasm, enabling the screening of more germplam as grain, before planting to the field, and thus enabling the efficient development of high %CP hemp grain populations.</p>
<p>For this study, a benchtop NIR spectrometer was used to develop a model to predict %CP content based on a data set of hemp grain representing multiple years, locations, and cultivars from grain and dual-purpose hemp types using PLSR.</p>
</section>
<section id="materials-and-methods" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="materials-and-methods"><span class="header-section-number">2</span> MATERIALS AND METHODS</h2>
<section id="hemp-grain-sample-background" class="level3" data-number="2.1">
<h3 data-number="2.1" class="anchored" data-anchor-id="hemp-grain-sample-background"><span class="header-section-number">2.1</span> Hemp Grain Sample Background</h3>
<p>Spectral data were obtained from whole (unground) hemp grain samples, harvested at maturity, collected from 2017 - 2021 from 18 cultivar trials in New York (NY) (149 samples). Grain samples were obtained by hand sampling or mechanical harvest and were cleaned of chaff and dried at 30 C for six days in a forced-air dryer. All %CP values are as percent dry matter. In total, 149 samples from 38 cultivars were represented in the data set. Cultivars were grain or dual-purpose types and included both commercially available and experimental material.</p>
<p>All cultivar trials were planted in randomized complete block design with each cultivar replicated four times. The 2017 data were comprised of samples from the same thirteen cultivars sampled from six NY locations. For those trials, grain was harvested from each plot individually and aggregated by cultivar within each trial. Four subsamples were drawn from each aggregated sample and scanned separately. These spectra were averaged at each 2 nm increment. All remaining samples from 2018-2021 were collected on a per-plot basis. All possible cultivars and possible locations were represented in 2017, but only a selected subset of cultivars and locations were represented in 2018-2021.</p>
</section>
<section id="spectral-data-collection-and-preprocessing" class="level3" data-number="2.2">
<h3 data-number="2.2" class="anchored" data-anchor-id="spectral-data-collection-and-preprocessing"><span class="header-section-number">2.2</span> Spectral Data Collection and Preprocessing</h3>
<p>A benchtop NIR spectrometer (FOSS/ NIR FOSS/ NIR Systems model 5000) was used to obtain the spectra (FOSS North America, Eden Prairie, MN, USA). Spectra were collected every 2 nm from 1100-2498 nm and the logarithm of reciprocal reflectance was recorded. A 1/4 rectangular sample cup (5.7 cm × 4.6 cm) was used to scan the samples.</p>
<p>WINISI software version 1.02A (Infrasoft International, Port Matilda, PA, USA) was used to average the spectra in 2017, as well as to select samples for laboratory assay. Samples were selected according to their spectral distance from their nearest neighbor within the calibration data set with a cutoff of a distance of 0.6 H, where H is approximately equal to the squared Mahalanobis distance divided by the number of principal components used in the calculation <span class="citation" data-cites="garrido-varo_note_2019">(<a href="#ref-garrido-varo_note_2019" role="doc-biblioref">Garrido-Varo et al., 2019</a>)</span>. Prior to selection selection, spectra were preprocessed using SNV-detrend with settings 1,4,4,1 for the derivative, gap, smooth, and smooth 2 settings respectively.</p>
</section>
<section id="laboratory-validation" class="level3" data-number="2.3">
<h3 data-number="2.3" class="anchored" data-anchor-id="laboratory-validation"><span class="header-section-number">2.3</span> Laboratory Validation</h3>
<p>Laboratory assays were performed by Dairy One Forage Laboratory (Ithaca, NY). For those assays, 1 mm ground samples were analyzed by combustion using a CN628 or CN928 Carbon/Nitrogen Determinator. Samples from 2017 were aggregated as described above, but the remaining samples were not aggregated.</p>
</section>
<section id="model-development" class="level3" data-number="2.4">
<h3 data-number="2.4" class="anchored" data-anchor-id="model-development"><span class="header-section-number">2.4</span> Model Development</h3>
<p>Training and testing sets were created by dividing the laboratory %CP values into tertiles according to their %CP in order to ensure that the range of %CP values was present in both calibration and testing sets. Within each tertile, 75% of the samples were randomly assigned to the training set and the remaining 25% were assigned to the testing set. For each training set, models were developed in the caret package using PLSR. In fitting the model, the number of components was optimized over a grid search from 1-20. Model performance was evaluated with 25 iterations of bootstrapping and minimized root mean squared error (RMSE) in selecting the number of components in the final model.</p>
<p>Initially a number of common spectral preprocessing methods were tested by creating 100 training and testing sets, as described above. Spectral data were transformed by each of the following methods: 1) first derivative, 2) Savitzky-Golay (SG) using the first derivative, third order polynomial, and a window of size 5, 3) gap-segment derivative using the first derivative, a gap of 11, and a segment size of 5, 4) standard normal variate (SNV), 5) standard normal variate following Savitzky-Golay (SNV-SG) using the same SG parameters as above, 6) SNV-detrend with second order polynomial, and 7) multiplicative scatter correction. For comparison, models were also developed using untransformed spectra.</p>
<p>For each of these preprocessing methods, models were fit and predictions were made on the corresponding testing set. Since there were 7 preprocessing methods as well as untransformed spectra, 8 separate models were fit for each of the 100 sets. The relationship between the predicted and actual values of the test set were calculated via RMSE, coefficient of determination (R<sup>2</sup>), relative predicted deviation (RPD), and Ratio of Performance to InterQuartile distance (RPIQ), four common model assessment metrics. Larger R<sup>2</sup>, RPD and RPIQ values and smaller RMSE values are best. The answer to the question of exactly which values constitute a “good” model varies depending upon the reference consulted, but for simplicity’s sake researchers desired a model with R<sup>2</sup> &gt; 0.80, an RPD greater than 2.5 and ideally greater than 3 (“good” to “excellent” quantitative prediction), and an RPIQ greater than 2.3 but ideally greater than 4.1 prediction on the testing set <span class="citation" data-cites="rawal_visible_2024 luce_prediction_2017 chadalavada_nir_2022">(<a href="#ref-chadalavada_nir_2022" role="doc-biblioref">Chadalavada et al., 2022</a>; <a href="#ref-luce_prediction_2017" role="doc-biblioref">Luce et al., 2017</a>; <a href="#ref-rawal_visible_2024" role="doc-biblioref">Rawal et al., 2024</a>)</span>.</p>
<p>Analyses of variance (ANOVA) were performed for each of these metrics in order to compare the preprocessing methods. For each ANOVA, each data set was considered as a subject and different variances were allowed for each preprocessing method. Once the most promising preprocessing method was identified, 1000 more training and testing sets were created and models were developed with that method. Performance on the testing sets was summarized with RMSE, R<sup>2</sup>, RPD, and RPIQ. The pattern of errors, expressed as the difference between the actual and predicted values for a given sample, was examined.</p>
</section>
<section id="additional-software-used" class="level3" data-number="2.5">
<h3 data-number="2.5" class="anchored" data-anchor-id="additional-software-used"><span class="header-section-number">2.5</span> Additional software used</h3>
<div class="cell">
<div class="cell-output-display">
<p>We used R version 4.3.3 <span class="citation" data-cites="base">(<a href="#ref-base" role="doc-biblioref">R Core Team, 2024</a>)</span> and the following R packages: caret v. 6.0.94 <span class="citation" data-cites="caret">(<a href="#ref-caret" role="doc-biblioref">Kuhn &amp; Max, 2008</a>)</span>, data.table v. 1.15.2 <span class="citation" data-cites="datatable">(<a href="#ref-datatable" role="doc-biblioref">Barrett et al., 2024</a>)</span>, emmeans v. 1.10.0 <span class="citation" data-cites="emmeans">(<a href="#ref-emmeans" role="doc-biblioref">Lenth, 2024</a>)</span>, nlme v. 3.1.163 <span class="citation" data-cites="nlme2000 nlme2023">(<a href="#ref-nlme2023" role="doc-biblioref">J. Pinheiro et al., 2023</a>; <a href="#ref-nlme2000" role="doc-biblioref">J. C. Pinheiro &amp; Bates, 2000</a>)</span>, pls v. 2.8.3 <span class="citation" data-cites="pls">(<a href="#ref-pls" role="doc-biblioref">Liland et al., 2023</a>)</span>, prospectr v. 0.2.7 <span class="citation" data-cites="prospectr">(<a href="#ref-prospectr" role="doc-biblioref">Stevens &amp; Ramirez-Lopez, 2024</a>)</span>, skimr v. 2.1.5 <span class="citation" data-cites="skimr">(<a href="#ref-skimr" role="doc-biblioref">Waring et al., 2022</a>)</span>, tidymodels v. 1.1.1 <span class="citation" data-cites="tidymodels">(<a href="#ref-tidymodels" role="doc-biblioref">Kuhn &amp; Wickham, 2020</a>)</span>, tidyverse v. 2.0.0 <span class="citation" data-cites="tidyverse">(<a href="#ref-tidyverse" role="doc-biblioref">Wickham et al., 2019</a>)</span>.</p>
</div>
<a class="quarto-notebook-link" id="nblink-1" href="index-preview.html">Source: Article Notebook</a></div>
</section>
</section>
<section id="results-and-discussion" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="results-and-discussion"><span class="header-section-number">3</span> RESULTS AND DISCUSSION</h2>
<section id="laboratory-assay-cp-values" class="level3" data-number="3.1">
<h3 data-number="3.1" class="anchored" data-anchor-id="laboratory-assay-cp-values"><span class="header-section-number">3.1</span> Laboratory assay %CP values</h3>
<p>Laboratory assay %CP values are summarized in the following table. These are similar to the range of %CP values observed in the literature, indicating an reasonable basis for a chemometric model. The %CP values were left-skewed (skewness of -0.29) and two thirds of the samples contained more than 25% CP.</p>
<div class="cell">
<div id="tbl-lab-protein-vals" class="cell quarto-float anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-lab-protein-vals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Summary of Laboratory Assayed CP Values (Percent Dry Matter)
</figcaption>
<div aria-describedby="tbl-lab-protein-vals-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="do-not-create-environment cell table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: right;">Mean</th>
<th style="text-align: right;">Sd</th>
<th style="text-align: right;">Minimum</th>
<th style="text-align: right;">First Quartile</th>
<th style="text-align: right;">Median</th>
<th style="text-align: right;">Third Quartile</th>
<th style="text-align: right;">Maximum</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">26.1</td>
<td style="text-align: right;">2.5</td>
<td style="text-align: right;">20.8</td>
<td style="text-align: right;">23.9</td>
<td style="text-align: right;">26.4</td>
<td style="text-align: right;">28.2</td>
<td style="text-align: right;">30.8</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>
</section>
<section id="preprocessing-methods-comparison" class="level3" data-number="3.2">
<h3 data-number="3.2" class="anchored" data-anchor-id="preprocessing-methods-comparison"><span class="header-section-number">3.2</span> Preprocessing methods comparison</h3>
<p>All preprocessing methods outperformed untransformed spectral data <a href="#tbl-preproc" class="quarto-xref">Table&nbsp;2</a>. Averaged together, all preprocessed spectra were superior to untransformed spectra, with lower RMSE and higher R<sup>2</sup>, RPD and RPIQ values (significant at <span class="math inline">\(\alpha\)</span> level &lt;0.001). Preprocessing methods had 11.6 % lower RMSE, and had 3.1% higher R<sup>2</sup>, 6.3% higher RPD and7.4% higher RPIQ than unprocessed spectra. Preprocessed spectra also had lower standard errors than untransformed spectra.</p>
<p>The SNV-SG method had the lowest RMSE and the highest R<sup>2</sup>, RPD and RPIQ averaging over all iterations. SNV-SG’s RMSE was 1.4% lower than the next best preprocessing method (SG), while SNV-SG’s R<sup>2</sup>, RPD, and RPIQ were 0.4%, 2.1%, and 2.4% higher than SG respectively. However, the differences between the best and second best methods by metric were only statistically significant at <span class="math inline">\(\alpha\)</span> &lt;0.05 for RPD and RPIQ. There is a long history of using RPD to evaluate chemometric models although the statistic has been criticized as inadequately reflecting the distribution of skewed populations, a situation which RPIQ was designed to address <span class="citation" data-cites="bellon-maurel_critical_2010">(<a href="#ref-bellon-maurel_critical_2010" role="doc-biblioref">Bellon-Maurel et al., 2010</a>)</span>. Here the data were somewhat but not heavily skewed and RPD and RPIQ metrics were in agreement. The superiority of SNV-SG by these metrics made it the best choice for the final model.</p>
<div class="cell">
<div id="tbl-preproc" class="cell quarto-float anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-preproc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2: Evaluation of Preprocessing Methods by Metric ± Standard Error
</figcaption>
<div aria-describedby="tbl-preproc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="do-not-create-environment cell table table-sm table-striped small">
<colgroup>
<col style="width: 48%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Preprocessing Method</th>
<th style="text-align: left;">RMSE</th>
<th style="text-align: left;"><span class="math inline">\(R^{2}\)</span></th>
<th style="text-align: left;">RPD</th>
<th style="text-align: left;">RPIQ</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Standard Normal Variate following Savitzky-Golay</td>
<td style="text-align: left;">1.02 ± 0.012</td>
<td style="text-align: left;">0.84 ± 0.004</td>
<td style="text-align: left;">2.49 ± 0.032</td>
<td style="text-align: left;">3.97 ± 0.076</td>
</tr>
<tr class="even">
<td style="text-align: left;">Savitzky-Golay</td>
<td style="text-align: left;">1.03 ± 0.012</td>
<td style="text-align: left;">0.83 ± 0.004</td>
<td style="text-align: left;">2.44 ± 0.029</td>
<td style="text-align: left;">3.88 ± 0.072</td>
</tr>
<tr class="odd">
<td style="text-align: left;">First Derivative</td>
<td style="text-align: left;">1.07 ± 0.013</td>
<td style="text-align: left;">0.82 ± 0.004</td>
<td style="text-align: left;">2.36 ± 0.032</td>
<td style="text-align: left;">3.77 ± 0.075</td>
</tr>
<tr class="even">
<td style="text-align: left;">Standard Normal Variate</td>
<td style="text-align: left;">1.12 ± 0.016</td>
<td style="text-align: left;">0.80 ± 0.005</td>
<td style="text-align: left;">2.26 ± 0.036</td>
<td style="text-align: left;">3.61 ± 0.081</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gap-segment Derivative</td>
<td style="text-align: left;">1.12 ± 0.018</td>
<td style="text-align: left;">0.81 ± 0.006</td>
<td style="text-align: left;">2.26 ± 0.040</td>
<td style="text-align: left;">3.60 ± 0.086</td>
</tr>
<tr class="even">
<td style="text-align: left;">Standard Normal Variate-Detrend</td>
<td style="text-align: left;">1.13 ± 0.015</td>
<td style="text-align: left;">0.80 ± 0.005</td>
<td style="text-align: left;">2.22 ± 0.035</td>
<td style="text-align: left;">3.55 ± 0.079</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Multiplicative Scatter Correction</td>
<td style="text-align: left;">1.17 ± 0.016</td>
<td style="text-align: left;">0.79 ± 0.006</td>
<td style="text-align: left;">2.17 ± 0.035</td>
<td style="text-align: left;">3.47 ± 0.080</td>
</tr>
<tr class="even">
<td style="text-align: left;">Untransformed Spectra</td>
<td style="text-align: left;">1.22 ± 0.044</td>
<td style="text-align: left;">0.79 ± 0.009</td>
<td style="text-align: left;">2.17 ± 0.052</td>
<td style="text-align: left;">3.42 ± 0.105</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>
<p>These results are not surprising. SNV and SNV-detrend both correct light scatter, which is often a function of differences in particle size and sample packing density, although SNV-detrend is often used for densely-packed, powdered samples <span class="citation" data-cites="barnes_standard_1989">(<a href="#ref-barnes_standard_1989" role="doc-biblioref">Barnes et al., 1989</a>)</span>. SG is a smoothing filter that regresses on the signal over a series of windows, removing noise while preserving the signal’s shape and features <span class="citation" data-cites="li_quantitative_2020 luo_properties_2005">(<a href="#ref-li_quantitative_2020" role="doc-biblioref">Li et al., 2020</a>; <a href="#ref-luo_properties_2005" role="doc-biblioref">Luo et al., 2005</a>)</span>. Derivatives, here including SG, gap-segment, and first derivatives pretreatments may remove additive and multiplicative effects, but not necessarily light scatter; as well, derivatives may increase spectral noise <span class="citation" data-cites="rinnan_review_2009">(<a href="#ref-rinnan_review_2009" role="doc-biblioref">Rinnan et al., 2009</a>)</span>. Here, hemp grain was neither powdered nor densely packed but samples were subject to light scatter and noise due to differences in particle size in the hemp grain.</p>
<p>The preprocessing methods examined represent a portion of those available. As well, these methods tend to have a number of user-adjustable parameters whose various permutations were not tested. This subset of preprocessing methods and parameters nonetheless contained substantial variations in model quality, demonstrating the importance of selecting an appropriate preprocessing method.</p>
</section>
<section id="final-model-development-and-summary" class="level3" data-number="3.3">
<h3 data-number="3.3" class="anchored" data-anchor-id="final-model-development-and-summary"><span class="header-section-number">3.3</span> Final model development and summary</h3>
<p>The model improved most rapidly as the number of components increased from 1 to 7, with the inclusion of each additional component being associated with a decrease in RMSE of 5-12% . From 8 to 12 components, model performance continued to improve, although gains were more modest (there was a decrease in RMSE of 0.7-3% with the inclusion of each additional component). With 13 or more components, performance gains were minimal and the relative ranks of the models tended to be stable <a href="#fig-model-calibration" class="quarto-xref">Figure&nbsp;1</a>.</p>
<div id="cell-fig-model-calibration" class="cell">
<div class="cell-output-display">
<div id="fig-model-calibration" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-model-calibration-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/fig-model-calibration-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: Decreasing RMSE with increasing number of components"><img src="index_files/figure-html/fig-model-calibration-1.png" class="img-fluid figure-img" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-model-calibration-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Decreasing RMSE with increasing number of components
</figcaption>
</figure>
</div>
</div>
</div>
<p>The final models performances on the test sets were similar, but not identical to, those obtained during the initial comparison of preprocessing methods. The final models’ mean RMSE was 1.03, R<sup>2</sup> was 0.83, RPD was 2.44, and RPIQ was 3.89. Five percent of the models were excellent by both metrics, with RPD &gt; 3 and RPIQ &gt; 4.1, while an additional 11% of the models were “good” by both metrics (RPD range from 2.5 - 3.0, RPIQ range from 2.3 - 4.1). Forty-nine percent of the models had the ability to approximate quantitative prediction (RPD range from 2.0 - 2.5), and 9% of the models were able to distinguish between higher and lower %CP values (RPD range from 1.5 - 2.0). Therefore, 74% of the models had, at minimum, the ability to distinguish between high and low values with with two thirds of them having, at minimum, the ability to approximate quantitative prediction. Despite the generally good model performance, a subset of poor models can be seen. For example, <a href="#fig-final-metric-boxplot" class="quarto-xref">Figure&nbsp;2</a> shows twenty-one models with R<sup>2</sup> below 0.7.</p>
<div id="cell-fig-final-metric-boxplot" class="cell">
<div class="cell-output-display">
<div id="fig-final-metric-boxplot" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-final-metric-boxplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/fig-final-metric-boxplot-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: Final model test set performance (1000 iterations)"><img src="index_files/figure-html/fig-final-metric-boxplot-1.png" class="img-fluid figure-img" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-final-metric-boxplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Final model test set performance (1000 iterations)
</figcaption>
</figure>
</div>
</div>
</div>
<p>Finally, the pattern of test set errors was examined on a per-sample basis by calculating the difference between the actual and predicted values for the samples in the test sets <a href="#fig-validation_set_performance" class="quarto-xref">Figure&nbsp;3</a> . A linear model was fit considering the mean estimated error for each sample where that sample was in the test set as compared to the sample’s actual value. The models overestimated %CP by approximately 0.5 % in the lowest tertile and underestimated %CP by -0.01 % and -0.41 % in the middle and highest tertile, respectively. The variance of the errors did not increase appreciably as %CP increased.</p>
<div id="cell-fig-validation_set_performance" class="cell">
<div class="cell-output-display">
<div id="fig-validation_set_performance" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-validation_set_performance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/fig-validation_set_performance-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: Test set prediction errors on a per-sample basis. Actual sample value set to 0, and samples ranked from least to greatest actual % CP value"><img src="index_files/figure-html/fig-validation_set_performance-1.png" class="img-fluid figure-img" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-validation_set_performance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Test set prediction errors on a per-sample basis. Actual sample value set to 0, and samples ranked from least to greatest actual % CP value
</figcaption>
</figure>
</div>
</div>
</div>
<p>The 15 (10%) best and 15 worst predicted samples as measured by the mean absolute error of prediction were identified and their backgrounds examined. Overall, half of the samples in the data set came from Ithaca, NY (“Ithaca”), while 28% were collected from Geneva, NY (“Geneva”) <a href="#tbl-hemp_provenance" class="quarto-xref">Table&nbsp;3</a>. However, of the 15 worst-predicted samples, 9 were from Geneva, while 3 of the 15 best-predicted samples were from Geneva (by contrast, 7 of the best-predicted and 5 of the worst-predicted samples came from Ithaca). Overall, samples from Geneva had the highest mean absolute error of prediction among locations, 61% greater than samples from Ithaca and 155% greater than samples from Freeville, NY (the only locations with more than 20 samples).</p>
<p>This study is limited in that it represents the creation of one model based upon spectra collected from one machine. NIRS calibrations can be unique to a particular machine, even if the machines compared are of the same model <span class="citation" data-cites="reeves_potential_2012">(<a href="#ref-reeves_potential_2012" role="doc-biblioref">Reeves, 2012</a>)</span>. As well, the calibration and validation sets are relatively small.</p>
<p>This research showed the promise of the use of NIRS in order to make predictions concerning %CP in hemp grain using PLS. Promising preprocessing methods were identified and a model was validated. Further research could refine a %CP model by including more samples, particularly by rectifying the class imbalance between Geneva and Ithaca, identifying promising spectral regions, or by examining other predictive methods.</p>
</section>
</section>
<section id="acknowledgments" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="acknowledgments"><span class="header-section-number">4</span> ACKNOWLEDGMENTS</h2>
<p>This work would not have been possible without the efforts of the field staff, undergraduate, and graduate students who planted, maintained, monitored and harvested these trials.</p>
</section>
<section id="conflict-of-interest" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="conflict-of-interest"><span class="header-section-number">5</span> CONFLICT OF INTEREST</h2>
<p>The authors declare no conflict of interest.</p>
</section>
<section id="orcid" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="orcid"><span class="header-section-number">6</span> ORCID</h2>
</section>
<section id="supplemental-material" class="level2" data-number="7">
<h2 data-number="7" class="anchored" data-anchor-id="supplemental-material"><span class="header-section-number">7</span> SUPPLEMENTAL MATERIAL</h2>
<div class="cell">
<div id="tbl-hemp_provenance" class="cell quarto-float anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-hemp_provenance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3: Tally of hemp cultivars and locations. Private cultivars are labeled “Cultivar1”, “Cultivar2”, etc., while experimental cultivars are labeled “Experimental1”, “Experimental2”, etc.
</figcaption>
<div aria-describedby="tbl-hemp_provenance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="do-not-create-environment cell table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Cultivar</th>
<th style="text-align: right;">Chazy</th>
<th style="text-align: right;">Freeville</th>
<th style="text-align: right;">Geneva</th>
<th style="text-align: right;">Ithaca</th>
<th style="text-align: right;">Willsboro</th>
<th style="text-align: right;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ALTAIR</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">ANKA</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BIALOBRZESKIE</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">CANDA</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CFX-1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">8</td>
</tr>
<tr class="even">
<td style="text-align: left;">CFX-2</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CRS-1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">CULTIVAR1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CULTIVAR2</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">CULTIVAR3</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CULTIVAR4</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">EARLINA 8</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">EXPERIMENTAL1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">EXPERIMENTAL2</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FELINA 32</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">FUTURA 75</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GRANDI</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">10</td>
</tr>
<tr class="even">
<td style="text-align: left;">H-51</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HAN-FN-H</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">HAN-NW</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HELENA</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">HENOLA</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">2</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HLESIA</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">3</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">HLIANA</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">JOEY</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">KATANI</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">9</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NEBRASKA (FERAL)</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">PEWTER RIVER</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PICOLO</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">8</td>
</tr>
<tr class="even">
<td style="text-align: left;">PORTUGAL</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ROCKY HEMP</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">STERLING GOLD</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SWIFT</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">TYGRA</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">USO-31</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">WOJKO</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">X-59</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">2</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;"></td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">TOTAL</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">41</td>
<td style="text-align: right;">76</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">149</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div>
</div>
</section>
<section id="optional-sections" class="level2" data-number="8">
<h2 data-number="8" class="anchored" data-anchor-id="optional-sections"><span class="header-section-number">8</span> OPTIONAL SECTIONS</h2>
</section>
<section id="references" class="level2" data-number="9">
<h2 data-number="9" class="anchored" data-anchor-id="references"><span class="header-section-number">9</span> REFERENCES</h2>
</section>
<section id="figures-and-tables" class="level2 unnumbered" data-number="10">


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">10 FIGURES AND TABLES</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-barnes_standard_1989" class="csl-entry" role="listitem">
Barnes, R. J., Dhanoa, M. S., &amp; Lister, S. J. (1989). Standard <span>Normal</span> <span>Variate</span> <span>Transformation</span> and <span>De</span>-<span>Trending</span> of <span>Near</span>-<span>Infrared</span> <span>Diffuse</span> <span>Reflectance</span> <span>Spectra</span>. <em>Applied Spectroscopy</em>, <em>43</em>(5), 772–777. <a href="https://doi.org/10.1366/0003702894202201">https://doi.org/10.1366/0003702894202201</a>
</div>
<div id="ref-datatable" class="csl-entry" role="listitem">
Barrett, T., Dowle, M., Srinivasan, A., Gorecki, J., Chirico, M., &amp; Hocking, T. (2024). <em><span class="nocase">data.table</span>: Extension of <span>“<span class="nocase">data.frame</span>”</span></em>. <a href="https://CRAN.R-project.org/package=data.table">https://CRAN.R-project.org/package=data.table</a>
</div>
<div id="ref-barta_proteomic_2024" class="csl-entry" role="listitem">
Bárta, J., Roudnický, P., Jarošová, M., Zdráhal, Z., Stupková, A., Bártová, V., Krejčová, Z., Kyselka, J., Filip, V., Říha, V., Lorenc, F., Bedrníček, J., &amp; Smetana, P. (2024). Proteomic <span>Profiles</span> of <span>Whole</span> <span>Seeds</span>, <span>Hulls</span>, and <span>Dehulled</span> <span>Seeds</span> of <span>Two</span> <span>Industrial</span> <span>Hemp</span> (<span>Cannabis</span> sativa <span>L</span>.) <span>Cultivars</span>. <em>Plants</em>, <em>13</em>(1), 111. <a href="https://doi.org/10.3390/plants13010111">https://doi.org/10.3390/plants13010111</a>
</div>
<div id="ref-bellon-maurel_critical_2010" class="csl-entry" role="listitem">
Bellon-Maurel, V., Fernandez-Ahumada, E., Palagos, B., Roger, J.-M., &amp; McBratney, A. (2010). Critical review of chemometric indicators commonly used for assessing the quality of the prediction of soil attributes by <span>NIR</span> spectroscopy. <em>TrAC Trends in Analytical Chemistry</em>, <em>29</em>(9), 1073–1081. <a href="https://doi.org/10.1016/j.trac.2010.05.006">https://doi.org/10.1016/j.trac.2010.05.006</a>
</div>
<div id="ref-callaway_hempseed_2004" class="csl-entry" role="listitem">
Callaway, J. C. (2004). Hempseed as a nutritional resource: <span>An</span> overview. <em>Euphytica</em>, <em>140</em>(1), 65–72. <a href="https://doi.org/10.1007/s10681-004-4811-6">https://doi.org/10.1007/s10681-004-4811-6</a>
</div>
<div id="ref-chadalavada_nir_2022" class="csl-entry" role="listitem">
Chadalavada, K., Anbazhagan, K., Ndour, A., Choudhary, S., Palmer, W., Flynn, J. R., Mallayee, S., Pothu, S., Prasad, K. V. S. V., Varijakshapanikar, P., Jones, C. S., &amp; Kholová, J. (2022). <span>NIR</span> <span>Instruments</span> and <span>Prediction</span> <span>Methods</span> for <span>Rapid</span> <span>Access</span> to <span>Grain</span> <span>Protein</span> <span>Content</span> in <span>Multiple</span> <span>Cereals</span>. <em>Sensors (Basel, Switzerland)</em>, <em>22</em>(10). <a href="https://doi.org/10.3390/s22103710">https://doi.org/10.3390/s22103710</a>
</div>
<div id="ref-ely_industrial_2022" class="csl-entry" role="listitem">
Ely, K., &amp; Fike, J. (2022). Industrial <span>Hemp</span> and <span>Hemp</span> <span>Byproducts</span> as <span>Sustainable</span> <span>Feedstuffs</span> in <span>Livestock</span> <span>Diets</span>. In D. C. Agrawal, R. Kumar, &amp; M. Dhanasekaran (Eds.), <em>Cannabis/<span>Hemp</span> for <span>Sustainable</span> <span>Agriculture</span> and <span>Materials</span></em> (pp. 145–162). Springer. <a href="https://doi.org/10.1007/978-981-16-8778-5_6">https://doi.org/10.1007/978-981-16-8778-5_6</a>
</div>
<div id="ref-garrido-varo_note_2019" class="csl-entry" role="listitem">
Garrido-Varo, A., Garcia-Olmo, J., &amp; Fearn, T. (2019). A note on <span>Mahalanobis</span> and related distance measures in <span>WinISI</span> and <span>The</span> <span>Unscrambler</span>. <em>Journal of Near Infrared Spectroscopy</em>, <em>27</em>(4), 253–258. <a href="https://doi.org/10.1177/0967033519848296">https://doi.org/10.1177/0967033519848296</a>
</div>
<div id="ref-geyer_genetics_2022" class="csl-entry" role="listitem">
Geyer, M., Mohler, V., &amp; Hartl, L. (2022). Genetics of the <span>Inverse</span> <span>Relationship</span> between <span>Grain</span> <span>Yield</span> and <span>Grain</span> <span>Protein</span> <span>Content</span> in <span>Common</span> <span>Wheat</span>. <em>Plants</em>, <em>11</em>(16), 2146. <a href="https://doi.org/10.3390/plants11162146">https://doi.org/10.3390/plants11162146</a>
</div>
<div id="ref-giancaspro_genetic_2019" class="csl-entry" role="listitem">
Giancaspro, A., Giove, S. L., Blanco, A., &amp; Gadaleta, A. (2019). Genetic <span>Variation</span> for <span>Protein</span> <span>Content</span> and <span>Yield</span>-<span>Related</span> <span>Traits</span> in a <span>Durum</span> <span>Population</span> <span>Derived</span> <span>From</span> an <span>Inter</span>-<span>Specific</span> <span>Cross</span> <span>Between</span> <span>Hexaploid</span> and <span>Tetraploid</span> <span>Wheat</span> <span>Cultivars</span>. <em>Frontiers in Plant Science</em>, <em>10</em>. <a href="https://doi.org/10.3389/fpls.2019.01509">https://doi.org/10.3389/fpls.2019.01509</a>
</div>
<div id="ref-hayes_measuring_2020" class="csl-entry" role="listitem">
Hayes, M. (2020). Measuring <span>Protein</span> <span>Content</span> in <span>Food</span>: <span>An</span> <span>Overview</span> of <span>Methods</span>. <em>Foods</em>, <em>9</em>(10), 1340. <a href="https://doi.org/10.3390/foods9101340">https://doi.org/10.3390/foods9101340</a>
</div>
<div id="ref-tidymodels" class="csl-entry" role="listitem">
Kuhn, M., &amp; Wickham, H. (2020). <em><span>Tidymodels</span>: A collection of packages for modeling and machine learning using tidyverse principles.</em> <a href="https://www.tidymodels.org">https://www.tidymodels.org</a>
</div>
<div id="ref-caret" class="csl-entry" role="listitem">
Kuhn, &amp; Max. (2008). Building predictive models in r using the caret package. <em>Journal of Statistical Software</em>, <em>28</em>(5), 1–26. <a href="https://doi.org/10.18637/jss.v028.i05">https://doi.org/10.18637/jss.v028.i05</a>
</div>
<div id="ref-emmeans" class="csl-entry" role="listitem">
Lenth, R. V. (2024). <em><span class="nocase">emmeans</span>: Estimated marginal means, aka least-squares means</em>. <a href="https://CRAN.R-project.org/package=emmeans">https://CRAN.R-project.org/package=emmeans</a>
</div>
<div id="ref-li_quantitative_2020" class="csl-entry" role="listitem">
Li, Y., Huang, Y., Xia, J., Xiong, Y., &amp; Min, S. (2020). Quantitative analysis of honey adulteration by spectrum analysis combined with several high-level data fusion strategies. <em>Vibrational Spectroscopy</em>, <em>108</em>, 103060. <a href="https://doi.org/10.1016/j.vibspec.2020.103060">https://doi.org/10.1016/j.vibspec.2020.103060</a>
</div>
<div id="ref-pls" class="csl-entry" role="listitem">
Liland, K. H., Mevik, B.-H., &amp; Wehrens, R. (2023). <em><span class="nocase">pls</span>: Partial least squares and principal component regression</em>. <a href="https://CRAN.R-project.org/package=pls">https://CRAN.R-project.org/package=pls</a>
</div>
<div id="ref-luce_prediction_2017" class="csl-entry" role="listitem">
Luce, M. S., Ziadi, N., Gagnon, B., &amp; Lévesque, V. (2017). Prediction of total carbon, total nitrogen, and <span class="nocase">pH</span> of organic materials using visible near-infrared reflectance spectroscopy. <em>Canadian Journal of Soil Science</em>, <em>98</em>(1), 175–179. <a href="https://doi.org/10.1139/cjss-2017-0109">https://doi.org/10.1139/cjss-2017-0109</a>
</div>
<div id="ref-luo_properties_2005" class="csl-entry" role="listitem">
Luo, J., Ying, K., He, P., &amp; Bai, J. (2005). Properties of <span>Savitzky</span>–<span>Golay</span> digital differentiators. <em>Digital Signal Processing</em>, <em>15</em>(2), 122–136. <a href="https://doi.org/10.1016/j.dsp.2004.09.008">https://doi.org/10.1016/j.dsp.2004.09.008</a>
</div>
<div id="ref-nlme2000" class="csl-entry" role="listitem">
Pinheiro, J. C., &amp; Bates, D. M. (2000). <em>Mixed-effects models in s and s-PLUS</em>. Springer. <a href="https://doi.org/10.1007/b98882">https://doi.org/10.1007/b98882</a>
</div>
<div id="ref-nlme2023" class="csl-entry" role="listitem">
Pinheiro, J., Bates, D., &amp; R Core Team. (2023). <em><span class="nocase">nlme</span>: Linear and nonlinear mixed effects models</em>. <a href="https://CRAN.R-project.org/package=nlme">https://CRAN.R-project.org/package=nlme</a>
</div>
<div id="ref-base" class="csl-entry" role="listitem">
R Core Team. (2024). <em><span>R</span>: A language and environment for statistical computing</em>. R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>
</div>
<div id="ref-rawal_visible_2024" class="csl-entry" role="listitem">
Rawal, A., Hartemink, A., Zhang, Y., Wang, Y., Lankau, R. A., &amp; Ruark, M. D. (2024). Visible and near-infrared spectroscopy predicted leaf nitrogen contents of potato varieties under different growth and management conditions. <em>Precision Agriculture</em>, <em>25</em>(2), 751–770. <a href="https://doi.org/10.1007/s11119-023-10091-z">https://doi.org/10.1007/s11119-023-10091-z</a>
</div>
<div id="ref-reeves_potential_2012" class="csl-entry" role="listitem">
Reeves, J. B. (2012). Potential of <span>Near</span>- and <span>Mid</span>-infrared <span>Spectroscopy</span> in <span>Biofuel</span> <span>Production</span>. <em>Communications in Soil Science and Plant Analysis</em>, <em>43</em>(1-2), 478–495. <a href="https://doi.org/10.1080/00103624.2012.641844">https://doi.org/10.1080/00103624.2012.641844</a>
</div>
<div id="ref-rinnan_review_2009" class="csl-entry" role="listitem">
Rinnan, Å., Berg, F. van den, &amp; Engelsen, S. B. (2009). Review of the most common pre-processing techniques for near-infrared spectra. <em>TrAC Trends in Analytical Chemistry</em>, <em>28</em>(10), 1201–1222. <a href="https://doi.org/10.1016/j.trac.2009.07.007">https://doi.org/10.1016/j.trac.2009.07.007</a>
</div>
<div id="ref-roberts_near-infrared_2004" class="csl-entry" role="listitem">
Roberts, C. A., Workman, J., &amp; Reeves, J. B. (2004). <em>Near-infrared spectroscopy in agriculture</em>. American Society of Agronomy.
</div>
<div id="ref-prospectr" class="csl-entry" role="listitem">
Stevens, A., &amp; Ramirez-Lopez, L. (2024). <em>An introduction to the prospectr package</em>.
</div>
<div id="ref-skimr" class="csl-entry" role="listitem">
Waring, E., Quinn, M., McNamara, A., Arino de la Rubia, E., Zhu, H., &amp; Ellis, S. (2022). <em><span class="nocase">skimr</span>: Compact and flexible summaries of data</em>. <a href="https://CRAN.R-project.org/package=skimr">https://CRAN.R-project.org/package=skimr</a>
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, H., Averick, M., Bryan, J., Chang, W., McGowan, L. D., François, R., Grolemund, G., Hayes, A., Henry, L., Hester, J., Kuhn, M., Pedersen, T. L., Miller, E., Bache, S. M., Müller, K., Ooms, J., Robinson, D., Seidel, D. P., Spinu, V., … Yutani, H. (2019). Welcome to the <span class="nocase">tidyverse</span>. <em>Journal of Open Source Software</em>, <em>4</em>(43), 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>
</div>
<div id="ref-williams_application_1975" class="csl-entry" role="listitem">
Williams, P. C. (1975). Application of near infrared reflectance spectroscopy to analysis of cereal grains and oilseeds. <em>Cereal Chemistry</em>, <em>52</em>(4 p.561-576), 576–561.
</div>
<div id="ref-wold_pls-regression_2001" class="csl-entry" role="listitem">
Wold, S., Sjöström, M., &amp; Eriksson, L. (2001). <span>PLS</span>-regression: A basic tool of chemometrics. <em>Chemometrics and Intelligent Laboratory Systems</em>, <em>58</em>(2), 109–130. <a href="https://doi.org/10.1016/S0169-7439(01)00155-1">https://doi.org/10.1016/S0169-7439(01)00155-1</a>
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"selector":".lightbox","loop":false,"closeEffect":"zoom","openEffect":"zoom","descPosition":"bottom"});
window.onload = () => {
  lightboxQuarto.on('slide_before_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    const href = trigger.getAttribute('href');
    if (href !== null) {
      const imgEl = window.document.querySelector(`a[href="${href}"] img`);
      if (imgEl !== null) {
        const srcAttr = imgEl.getAttribute("src");
        if (srcAttr && srcAttr.startsWith("data:")) {
          slideConfig.href = srcAttr;
        }
      }
    } 
  });

  lightboxQuarto.on('slide_after_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    if (window.Quarto?.typesetMath) {
      window.Quarto.typesetMath(slideNode);
    }
  });

};
          </script>




</body></html>