<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.5.24">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

    <meta name="author" content="Ryan Crawford">
    <meta name="author" content="Jamie Crawford">
    <meta name="author" content="Lawrence B. Smart">
    <meta name="author" content="Virginia Moore">
    <meta name="dcterms.date" content="2024-03-12">
    <meta name="keywords" content="Hemp, Grain, Spectroscopy">

    <title>Near Infra-Red Spectroscopy Predicts Crude Protein in Hemp Grain</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
      /* CSS for citations */
      div.csl-bib-body { }
      div.csl-entry {
        clear: both;
        margin-bottom: 0em;
      }
      .hanging-indent div.csl-entry {
        margin-left:2em;
        text-indent:-2em;
      }
      div.csl-left-margin {
        min-width:2em;
        float:left;
      }
      div.csl-right-inline {
        margin-left:2em;
        padding-left:1em;
      }
      div.csl-indent {
        margin-left: 2em;
      }    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>   <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script> 
      <meta name="citation_title" content="Near Infra-Red Spectroscopy Predicts Crude Protein in Hemp Grain">
<meta name="citation_abstract" content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
">
<meta name="citation_keywords" content="Hemp,Grain,Spectroscopy">
<meta name="citation_author" content="Ryan Crawford">
<meta name="citation_author" content="Jamie Crawford">
<meta name="citation_author" content="Lawrence B. Smart">
<meta name="citation_author" content="Virginia Moore">
<meta name="citation_publication_date" content="2024-03-12">
<meta name="citation_cover_date" content="2024-03-12">
<meta name="citation_year" content="2024">
<meta name="citation_online_date" content="2024-03-12">
<meta name="citation_language" content="en">
<meta name="citation_reference" content="citation_title=NIR Instruments and Prediction Methods for Rapid Access to Grain Protein Content in Multiple Cereals.;,citation_abstract=Achieving global goals for sustainable nutrition, health, and wellbeing will depend on delivering enhanced diets to humankind. This will require instantaneous access to information on food-source quality at key points of agri-food systems. Although laboratory analysis and benchtop NIR spectrometers are regularly used to quantify grain quality, these do not suit all end users, for example, stakeholders in decentralized agri-food chains that are typical in emerging economies. Therefore, we explored benchtop and portable NIR instruments, and the methods that might aid these particular end uses. For this purpose, we generated NIR spectra for 328 grain samples from multiple cereals (finger millet, foxtail millet, maize, pearl millet, and sorghum) with a standard benchtop NIR spectrometer (DS2500, FOSS) and a novel portable NIR-based instrument (HL-EVT5, Hone). We explored classical deterministic methods (via winISI, FOSS), novel machine learning (ML)-driven methods (via Hone Create, Hone), and a convolutional neural network (CNN)-based method for building the calibrations to predict grain protein out of the NIR spectra. All of the tested methods enabled us to build relevant calibrations out of both types of spectra (i.e., R ^{\textrm{2}} ≥ 0.90, RMSE ≤ 0.91, RPD ≥ 3.08). Generally, the calibration methods integrating the ML techniques tended to enhance the prediction capacity of the model. We also documented that the prediction of grain protein content based on the NIR spectra generated using the novel portable instrument (HL-EVT5, Hone) was highly relevant for quantitative protein predictions (R ^{\textrm{2}} = 0.91, RMSE = 0.97, RPD = 3.48). Thus, the presented findings lay the foundations for the expanded use of NIR spectroscopy in agricultural research, development, and trade.;,citation_author=Keerthi Chadalavada;,citation_author=Krithika Anbazhagan;,citation_author=Adama Ndour;,citation_author=Sunita Choudhary;,citation_author=William Palmer;,citation_author=Jamie R Flynn;,citation_author=Srikanth Mallayee;,citation_author=Sharada Pothu;,citation_author=Kodukula Venkata Subrahamanya Vara Prasad;,citation_author=Padmakumar Varijakshapanikar;,citation_author=Chris S Jones;,citation_author=Jana Kholová;,citation_publication_date=2022-05;,citation_cover_date=2022-05;,citation_year=2022;,citation_issue=10;,citation_doi=10.3390/s22103710;,citation_issn=1424-8220;,citation_volume=22;,citation_journal_title=Sensors (Basel, Switzerland);">
<meta name="citation_reference" content="citation_title=Estimation of crude protein and amino acid contents in whole, ground and defatted ground soybeans by different types of near-infrared (NIR) reflectance spectroscopy;,citation_abstract=This study was designed to examine the ability of near-infrared (NIR) spectroscopy to determine amino acid and crude protein concentrations in soybean and explore the effects of particle size, fat content and types of NIR spectroscopy instruments on the predictive ability of the NIR models. Whole, ground and defatted ground soybean were scanned by a diode array NIR analyzer and defatted ground soybean were scanned by a Fourier transformed NIR analyzer. Samples were analyzed for crude protein and amino acid contents using reference wet chemistry methods. The NIR calibration models of crude protein and most amino acids showed acceptable coefficient of determinations (R2c = 0.605–0.952), while the NIR models for Cys, Met and Trp were less accurate (R2c = 0.498–0.667 for Cys and R2c = 0.482–0.615 for Met and R2c = 0.406–0.481 for Trp). The R2c values in NIR calibration were found to relate to the correlation between amino acids and crude protein. The grinding process and lipid extraction were found to improve the NIR spectroscopy in crude protein and amino acids prediction. The Fourier transformed NIR analyzer was found to perform similar to the diode array NIR analyzer for defatted ground soybean analysis. No significant differences were found between crude protein and amino acid contents predicted by these two NIR spectroscopy instruments (P&nbsp;&amp;amp;amp;gt;&nbsp;0.05).;,citation_author=Da Shi;,citation_author=Jiayi Hang;,citation_author=Jason Neufeld;,citation_author=Shusheng Zhao;,citation_author=James D. House;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0889157522002198;,citation_doi=10.1016/j.jfca.2022.104601;,citation_issn=0889-1575;,citation_volume=111;,citation_journal_title=Journal of Food Composition and Analysis;">
<meta name="citation_reference" content="citation_title=Comparison of PLSR, MLR, SVM regression methods for determination of crude protein and carbohydrate content in stored wheat using near Infrared spectroscopy;,citation_abstract=Aim of the study was to predict quality parameters namely protein and carbohydrate content of wheat grain stored for one year at 4&nbsp;°C and 21&nbsp;°C using Near Infrared spectroscopy (NIRS) and Chemometrics. The spectra of grains were measured in reflectance mode using lab made filter based pre dispersive NIRS ranging from wavelength 750&nbsp;nm to 2580&nbsp;nm. NIR Wavelengths were divided into two sets for all the stored samples depending on its correlation with factors. The reference data and NIR data were analysed by principal component analysis (PCA) scores, partial last squared regression (PLSR) model, Multiple Regression Analysis (MLR) and Support Vector Machine Regression (SVMR). First four factors were sufficient to develop regression models for the above mentioned methods. Prediction performance was compared on the basis of the coefficient of correlation (R2) and Root Mean Square Error (RMSE) for the calibration and validation sets. R2 values for prediction of protein content from PLSR model were 0.955, 0.997 for 4&nbsp;°C Wavelength Set I and II and 0.903, 0.989 for 21&nbsp;°C Wavelength Set I and II. For the prediction of carbohydrate content 0.978, 0.951 and 0.946, 0.974 were the R2 values for the 4&nbsp;°C and 21&nbsp;°C wheat samples Set I and Set II respectively. MLR model also gave R2 values greater than 0.9 for all the samples. Wavelengths with high β correlation coefficients were defined. SVMR predicted the protein content and carbohydrate content with good accuracy. SVMR and PLS results. All the models developed showed good accuracy for successful prediction of protein and carbohydrate content of stored wheat grains with PLSR and SVMR giving the best results. This study showed that near infrared spectroscopy has potential to distinguish wheat grains stored under different conditions.;,citation_author=Uma Kamboj;,citation_author=Paramita Guha;,citation_author=Sunita Mishra;,citation_publication_date=2022-01;,citation_cover_date=2022-01;,citation_year=2022;,citation_doi=10.1016/j.matpr.2021.04.540;,citation_issn=2214-7853;,citation_volume=48;,citation_journal_title=Materials Today: Proceedings;">
<meta name="citation_reference" content="citation_title=Intelligent evaluation of free amino acid and crude protein content in raw peanut seed kernels using NIR spectroscopy paired with multivariable calibration;,citation_abstract=Given the nutritional importance of peanuts, this study examined the free amino acid (FAA) and crude protein (CP) content in raw peanut seeds. Near-infrared spectroscopy (NIRS) was employed in combination with variable selection algorithms after successful reference data analysis using colorimetric and Kjeldahl methods. Ensuing the application of partial least squares (PLS) as a full spectral model, the genetic algorithm (GA), bootstrapping soft shrinkage (BOSS), uninformative variable elimination (UVE), and random frog (RF) models were tested and assessed. A comparison of correlation coefficients of prediction (Rₚ), root mean square error of prediction (RMSEP), and residual predictive deviation (RPD) was performed to appraise the performance of the built models. Using RF-PLS, an unsurpassed outcome was achieved for FAA (Rₚ = 0.937, RPD = 3.38) and CP (Rₚ = 0.9261, RPD = 3.66). These findings demonstrated that NIR in combination with RF-PLS could be utilized for quantitative, rapid, and nondestructive prediction of FAA and CP in raw peanut seed samples.;,citation_author=Suleiman A. Haruna;,citation_author=Huanhuan Li;,citation_author=Wenya Wei;,citation_author=Wenhui Geng;,citation_author=Selorm Yao-Say Solomon Adade;,citation_author=Muhammad Zareef;,citation_author=Ngouana Moffo A. Ivane;,citation_author=Quansheng Chen;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_issue=31;,citation_doi=10.1039/d2ay00875k;,citation_issn=1759-9679;,citation_volume=14;,citation_journal_title=Analytical methods;">
<meta name="citation_reference" content="citation_title=High throughput phenotyping of Camelina sativa seeds for crude protein, total oil, and fatty acids profile by near infrared spectroscopy;,citation_abstract=Highlights •Near infrared spectroscopy (NIRS) is a fast, non-destructive method for high throughput phenotyping of camelina seed.•NIRS works well for evaluating camelina seed crude protein, total oil, and fatty acid profiles.•NIRS calibration equations for phenotyping agronomic traits of camelina seed will be useful for conducting genome wide association studies.;,citation_author=James V. Anderson;,citation_author=Alex Wittenberg;,citation_author=Hui Li;,citation_author=Marisol T. Berti;,citation_publication_date=2019-10;,citation_cover_date=2019-10;,citation_year=2019;,citation_doi=10.1016/j.indcrop.2019.04.075;,citation_issn=0926-6690;,citation_volume=137;,citation_journal_title=Industrial Crops &amp;amp;amp; Products;">
<meta name="citation_reference" content="citation_title=Accuracy of NIRS Instruments to Analyze Forage and Grain1;,citation_abstract=A study was made of the relative accuracy of three near infrared reflectance (NIR) instruments: a monochromator and two lower cost scanning filter instruments. All instruments used in this test were manufactured by Pacific Scientific, Silver Spring, MD. The monochromator was the 6100 with a scanning range of 1100 to 2498 nm. The 51A had a scanning range from 1596 to 1756 nm and 1988 to 2356 nm. The 102 scanned from 1902 to 2320 nm. Computer programs developed for the National NIR Forage Project were used to conduct the instrument comparisons tests. Agricultural products used in the experiment were hay, haylage, corn (Zea mays L.) silage and corn grain. Analyses included crude protein (CP), acid detergent fiber (ADF), Ca, P, K, M, and dry matter (DM). The 6100 generally had lower standard errors of validation than the scanning filter instruments. Although the six-filter 51A had more spectral information than the three-filter 102, it did not consistently out-perform the 102. There was no trend for one math treatment or series of wavelengths to be selected across instruments, products, or analyses. Regardless of the instrument spectral capabilities, the fraction of explained variance (R2) values for CP, ADF, and DM were higher than the R2s for the minerals. Even though the accuracy of the filter instruments was often below that of the scanning monochromator, the differences were small and would be of little significance for many agricultural applications.;,citation_author=John S. Shenk;,citation_author=Mark O. Westerhaus;,citation_publication_date=1985;,citation_cover_date=1985;,citation_year=1985;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.2135/cropsci1985.0011183X002500060054x;,citation_issue=6;,citation_doi=10.2135/cropsci1985.0011183X002500060054x;,citation_issn=1435-0653;,citation_volume=25;,citation_journal_title=Crop Science;">
<meta name="citation_reference" content="citation_title=Moisture effects on robustness of sorghum grain protein near‐infrared spectroscopy calibration;,citation_abstract=BACKGROUND AND OBJECTIVES: A near‐infrared (NIR) spectroscopy method was developed for rapid and nondestructive evaluation of protein content of intact sorghum grains. Effect of grain sample moisture variation on the robustness of protein calibration was investigated. FINDINGS: An initial NIR protein calibration model with sorghum grains in 7.28%–11.57% moisture content range with coefficient of determination (R²) of 0.94 and standard error of cross‐validation (SECV) of 0.41%, predicted protein content of an external validation set of different varieties with R²&nbsp;=&nbsp;0.90, root‐mean‐square error of prediction (RMSEP)&nbsp;=&nbsp;0.42% and bias&nbsp;=&nbsp;−0.02%. However, when grains with a wider range of moisture content (7.50%–17.75%) were used for validation, prediction errors increased with R²&nbsp;=&nbsp;0.72 RMSEP&nbsp;=&nbsp;0.87% and bias&nbsp;=&nbsp;−0.32%. Inclusion of grains with a wider moisture range to the calibration set improved the performance of the calibration model with a R²&nbsp;=&nbsp;0.83, RMSEP&nbsp;=&nbsp;0.67% with a bias of −0.04%. CONCLUSIONS: Variation of moisture content in grains affected the performance of the NIR protein calibration model. Likewise, inclusion of moisture variation in the calibration sample set improved the robustness of the model. SIGNIFICANCE AND NOVELTY: In addition to the traits of interest, variation of other physical or chemical traits should also be considered for inclusion into the calibration sample set to improve the robustness of NIR calibration models. This is especially important when NIR spectroscopy methods are developed for evaluation of breeding populations as the future grain samples from numerous crosses may be substantially diverse.;,citation_author=Kamaranga H. S. Peiris;,citation_author=Scott R. Bean;,citation_author=Anuj Chiluwal;,citation_author=Ramasamy Perumal;,citation_author=S. V. Krishna Jagadish;,citation_publication_date=2019-01;,citation_cover_date=2019-01;,citation_year=2019;,citation_issue=4 p.678-688;,citation_doi=10.1002/cche.10164;,citation_issn=00090352;,citation_volume=96;,citation_journal_title=Cereal chemistry;">
<meta name="citation_reference" content="citation_title=Potential of Near- and Mid-infrared Spectroscopy in Biofuel Production;,citation_abstract=For many decades, near-infrared spectroscopy (NIRS) has been used to determine the composition of animal feedstuffs and grains. More recently, mid-infrared spectroscopy (mid-IR) has also been examined for similar determinations. These spectroscopic methods offer the potential for rapid and accurate determination of organic constituents, such as fiber components and protein, of forages, by-products, and grains at reduced cost and greatly increased speed (minutes instead of hours or days). Because they are nonchemical in nature, they result in a large reduction (90% or more) in the chemical wastes associated with standard chemical-based assay methods. The same components of interest for biofuel production (cellulose, hemicellulose, lignin, starch, protein, oil, etc.) are those that have already been determined by NIRS/mid-IR for evaluating grains and animal feedstuffs. Therefore, these techniques would appear to be a natural match for evaluating feedstocks for biofuels, and the literature shows that efforts in this direction are being successfully tested and instituted. For this discussion, an overview of where such efforts are and the potential for NIRS/mid-IR in producing biofuels will be covered. For example, while there are similarities between the needs of the biofuels industry and the analysis of animal feedstuffs, there are also both practical and technical differences between the two that will likely impact how NIRS/mid-IR is developed for biofuels. As an example, grain analysis for protein is performed on a large scale by government agencies such as the Canadian Grain Commission and U.S. Grain Inspection Service, while at least in the United States, animal feedstuff analysis is performed by state or independent laboratories for individual farmers. For biofuels, this might well result in most analysis being performed by the large corporations converting the feedstocks to biofuels, as opposed to the individual producer having analysis performed at an independent laboratory. Similarly for animal feedstuffs, measurements of fiber (neutral detergent fiber or NDF, acid detergent fiber or ADF, and lignin) and protein are carried out. These fiber measurements often consist of more than one type of fiber component with some being computed by difference (hemicellulose = NDF – ADF) and are empirical at best. Whether such empirical estimates will be sufficient for assessing biofuels or whether new spectroscopic methods for directly measuring the components of interest (cellulose, etc.) will need to be developed is a question to be answered when components other than starch for ethanol or oil for biodiesel become common.;,citation_author=James B. Reeves;,citation_publication_date=2012-01;,citation_cover_date=2012-01;,citation_year=2012;,citation_fulltext_html_url=https://doi.org/10.1080/00103624.2012.641844;,citation_issue=1-2;,citation_doi=10.1080/00103624.2012.641844;,citation_issn=0010-3624;,citation_volume=43;,citation_journal_title=Communications in Soil Science and Plant Analysis;">
<meta name="citation_reference" content="citation_title=Near-infrared spectroscopy in agriculture;,citation_author=Craig A Roberts;,citation_author=Jerry Workman;,citation_author=James B Reeves;,citation_publication_date=2004;,citation_cover_date=2004;,citation_year=2004;">
<meta name="citation_reference" content="citation_title=A note on Mahalanobis and related distance measures in WinISI and The Unscrambler;,citation_abstract=In identifying spectral outliers in near infrared calibration it is common to use a distance measure that is related to Mahalanobis distance. However, different software packages tend to use different variants, which lead to a translation problem if more than one package is used. Here the relationships between squared Mahalanobis distance D2, the GH distance of WinISI, and the T2 and leverage (L) statistics of The Unscrambler are established as D2?=?T2 ≈ L???n ≈ GH???k, where n and k are the numbers of samples and variables, respectively, in the set of spectral data used to establish the distance measure. The implications for setting thresholds for outlier detection are discussed. On the way to this result the principal component scores from WinISI and The Unscrambler are compared. Both packages scale the scores for a component to have variances proportional to the contribution of that component to total variance, but the WinISI scores, unlike those from The Unscrambler, do not have mean zero.;,citation_author=A Garrido-Varo;,citation_author=J Garcia-Olmo;,citation_author=T Fearn;,citation_publication_date=2019-08;,citation_cover_date=2019-08;,citation_year=2019;,citation_fulltext_html_url=https://doi.org/10.1177/0967033519848296;,citation_issue=4;,citation_doi=10.1177/0967033519848296;,citation_issn=0967-0335;,citation_volume=27;,citation_journal_title=Journal of Near Infrared Spectroscopy;">
<meta name="citation_reference" content="citation_title=Application of near infrared reflectance spectroscopy to analysis of cereal grains and oilseeds;,citation_author=P. C. Williams;,citation_publication_date=1975-07;,citation_cover_date=1975-07;,citation_year=1975;,citation_issue=4 p.561-576;,citation_issn=0009-0352;,citation_volume=52;,citation_journal_title=Cereal chemistry;">
<meta name="citation_reference" content="citation_title=The impact of hempseed dehulling on chemical composition, structure properties and aromatic profile of hemp protein isolate;,citation_abstract=Hempseed protein has attracted increasing attention in both scientific and industrial field because of its high nutritional value. In the current study, the influence of dehulling process on the chemical composition, structure and functional properties of hempseed protein isolate (HPI) was determined by using SDS-PAGE, SEC-HPLC-UV/RI/MALS, CD spectroscopy and DSC. The result demonstrated that dehulling process significantly increased the extraction and protein recovery yields of HPI. Meanwhile, dehulled HPI also exhibited a higher protein purity and Arg/Lys ratio with improved color. No significant impact on protein composition and structure was observed upon seed dehulling, but HPI extracted from the dehulled seeds showed a reduced thermal stability as compared to non-dehulled HPI. The aromatic profile of HPI was determined by HS-SPME-GC-MS, and the data suggested that dehulled HPI accumulated more terpenes and less lipid oxidation volatiles than non-dehulled HPI. These findings indicated that dehulling process greatly enhances the extraction yield with higher protein content and more characteristic aroma of HPI. The current study for the first time provides valuable information about the influence of dehulling processing on the properties of HPI, which will facilitate further research and application of hempseed and its valued added ingredients in both academic and industry field.;,citation_author=Peiyi Shen;,citation_author=Zili Gao;,citation_author=Minwei Xu;,citation_author=Jae-Bom Ohm;,citation_author=Jiajia Rao;,citation_author=Bingcan Chen;,citation_publication_date=2020-09;,citation_cover_date=2020-09;,citation_year=2020;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0268005X20301338;,citation_doi=10.1016/j.foodhyd.2020.105889;,citation_issn=0268-005X;,citation_volume=106;,citation_journal_title=Food Hydrocolloids;">
<meta name="citation_reference" content="citation_title=Genotype x Environmental Effects on Yielding Ability and Seed Chemical Composition of Industrial Hemp (Cannabis sativa L.) Varieties Grown in North Dakota, USA;,citation_abstract=The influence of growing conditions in North Dakota on grain yield and seed composition was evaluated for industrial hemp varieties (Cannabis sativa L.) grown in North Dakota, USA. Average grain yield across varieties was 2138 kg ha−1. Crude protein content of undehulled industrial hemp flour from 10 varieties ranged from 32.7% to 35.9%; and oil content ranged from 24.3% to 28.1%. All industrial hemp flours were excellent natural sources of iron (46.7%), manganese (169.1%), copper (29.0%), zinc (28.2%), phosphorus (41.0%), and magnesium (33.7%) on a per serving basis (30 g per serving) for percent recommended daily intake (% RDI). Oils from most of the varieties differed significantly for the fatty acid (FA) profile. The n6/n3 ratio ranged from 3.2 to 5.1. Principal component analysis (PCA) suggested that CFX-2 and CRS-1 grew well in North Dakota based on grain yield, flour, and oil quality. In this study, the chemical composition of flour was not influenced by crop year; but crop year did effect the nutritional mineral composition. These results may have important implications for the use of industrial hemp in food applications.;,citation_author=Yang Lan;,citation_author=Fengchao Zha;,citation_author=Allen Peckrul;,citation_author=Bryan Hanson;,citation_author=Burton Johnson;,citation_author=Jiajia Rao;,citation_author=Bingcan Chen;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1002/aocs.12291;,citation_issue=12;,citation_doi=10.1002/aocs.12291;,citation_issn=1558-9331;,citation_volume=96;,citation_journal_title=Journal of the American Oil Chemists’ Society;">
<meta name="citation_reference" content="citation_title=Hempseed as a nutritional resource: An overview;,citation_abstract=The seed of Cannabis sativa L. has been an important source of nutrition for thousands of years in Old World cultures. Non-drug varieties of Cannabis, commonly referred to as hemp, have not been studied extensively for their nutritional potential in recent years, nor has hempseed been utilized to any great extent by the industrial processes and food markets that have developed during the 20th century. Technically a nut, hempseed typically contains over 30% oil and about 25% protein, with considerable amounts of dietary fiber, vitamins and minerals. Hempseed oil is over 80% in polyunsaturated fatty acids (PUFAs), and is an exceptionally rich source of the two essential fatty acids (EFAs) linoleic acid (18:2 omega-6) and alpha-linolenic acid (18:3 omega-3). The omega-6 to omega-3 ratio (n6/n3) in hempseed oil is normally between 2:1 and 3:1, which is considered to be optimal for human health. In addition, the biological metabolites of the two EFAs, gamma-linolenic acid (18:3 omega-6; “GLA”) and stearidonic acid (18:4 omega-3; “SDA”), are also present in hempseed oil. The two main proteins in hempseed are edestin and albumin. Both of these high-quality storage proteins are easily digested and contain nutritionally significant amounts of all essential amino acids. In addition, hempseed has exceptionally high levels of the amino acid arginine. Hempseed has been used to treat various disorders for thousands of years in traditional oriental medicine. Recent clinical trials have identified hempseed oil as a functional food, and animal feeding studies demonstrate the long-standing utility of hempseed as an important food resource.;,citation_author=J. C. Callaway;,citation_publication_date=2004-01;,citation_cover_date=2004-01;,citation_year=2004;,citation_fulltext_html_url=https://doi.org/10.1007/s10681-004-4811-6;,citation_issue=1;,citation_doi=10.1007/s10681-004-4811-6;,citation_issn=1573-5060;,citation_volume=140;,citation_journal_title=Euphytica;">
<meta name="citation_reference" content="citation_title=Proteomic Profiles of Whole Seeds, Hulls, and Dehulled Seeds of Two Industrial Hemp (Cannabis sativa L.) Cultivars;,citation_abstract=As a source of nutritionally important components, hemp seeds are often dehulled for consumption and food applications by removing the hard hulls, which increases their nutritional value. The hulls thus become waste, although they may contain valuable protein items, about which there is a lack of information. The present work is therefore aimed at evaluating the proteome of hemp (Cannabis sativa L.) at the whole-seed, dehulled seed, and hull levels. The evaluation was performed on two cultivars, Santhica 27 and Uso-31, using LC-MS/MS analysis. In total, 2833 protein groups (PGs) were identified, and their relative abundances were determined. A set of 88 PGs whose abundance exceeded 1000 ppm (MP88 set) was considered for further evaluation. The PGs of the MP88 set were divided into ten protein classes. Seed storage proteins were found to be the most abundant protein class: the averages of the cultivars were 65.5%, 71.3%, and 57.5% for whole seeds, dehulled seeds, and hulls, respectively. In particular, 11S globulins representing edestin (three PGs) were found, followed by 7S vicilin-like proteins (four PGs) and 2S albumins (two PGs). The storage 11S globulins in Santhica 27 and Uso-31 were found to have a higher relative abundance in the dehulled seed proteome (summing to 58.6 and 63.2%) than in the hull proteome (50.5 and 54%), respectively. The second most abundant class of proteins was oleosins, which are part of oil-body membranes. PGs belonging to metabolic proteins (e.g., energy metabolism, nucleic acid metabolism, and protein synthesis) and proteins related to the defence and stress responses were more abundant in the hulls than in the dehulled seeds. The hulls can, therefore, be an essential source of proteins, especially for medical and biotechnological applications. Proteomic analysis has proven to be a valuable tool for studying differences in the relative abundance of proteins between dehulled hemp seeds and their hulls among different cultivars.;,citation_author=Jan Bárta;,citation_author=Pavel Roudnický;,citation_author=Markéta Jarošová;,citation_author=Zbyněk Zdráhal;,citation_author=Adéla Stupková;,citation_author=Veronika Bártová;,citation_author=Zlatuše Krejčová;,citation_author=Jan Kyselka;,citation_author=Vladimír Filip;,citation_author=Václav Říha;,citation_author=František Lorenc;,citation_author=Jan Bedrníček;,citation_author=Pavel Smetana;,citation_publication_date=2024-01;,citation_cover_date=2024-01;,citation_year=2024;,citation_fulltext_html_url=https://www.mdpi.com/2223-7747/13/1/111;,citation_issue=1;,citation_doi=10.3390/plants13010111;,citation_issn=2223-7747;,citation_volume=13;,citation_journal_title=Plants;">
<meta name="citation_reference" content="citation_title=Genetics of the Inverse Relationship between Grain Yield and Grain Protein Content in Common Wheat;,citation_abstract=Grain protein content (GPC) is one of the most important criteria to determine the quality of common wheat (Triticum aestivum). One of the major obstacles for bread wheat production is the negative correlation between GPC and grain yield (GY). Previous studies demonstrated that the deviation from this inverse relationship is highly heritable. However, little is known about the genetics controlling these deviations in common wheat. To fill this gap, we performed quantitative trait locus (QTL) analysis for GY, GPC, and four derived GY-GPC indices using an eight-way multiparent advanced generation intercross population comprising 394 lines. Interval mapping was conducted using phenotypic data from up to nine environments and genotypic data from a 20k single-nucleotide polymorphism array. The four indices were highly heritable (0.76–0.88) and showed distinct correlations to GY and GPC. Interval mapping revealed that GY, GPC, and GY-GPC indices were controlled by 6, 12, and 12 unique QTL, of which each explained only a small amount of phenotypic variance (R2 ≤ 10%). Ten of the 12 index QTL were independent of loci affecting GY and GPC. QTL regions harboured several candidate genes, including Rht-1, WAPO-A1, TaTEF-7A, and NRT2.6-7A. The study confirmed the usefulness of indices to mitigate the inverse GY-GPC relationship in breeding, though the selection method should reflect their polygenic inheritance.;,citation_author=Manuel Geyer;,citation_author=Volker Mohler;,citation_author=Lorenz Hartl;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9413592/;,citation_issue=16;,citation_doi=10.3390/plants11162146;,citation_issn=2223-7747;,citation_pmid=36015449;,citation_volume=11;,citation_journal_title=Plants;">
<meta name="citation_reference" content="citation_title=Genetic Variation for Protein Content and Yield-Related Traits in a Durum Population Derived From an Inter-Specific Cross Between Hexaploid and Tetraploid Wheat Cultivars;,citation_abstract=&amp;amp;amp;lt;p&amp;gt;Wheat grain protein content (GPC) and yield components are complex quantitative traits influenced by a multi-factorial system consisting of both genetic and environmental factors. Although seed storage proteins represent less than 15% of mature kernels, they are crucial in determining end-use properties of wheat, as well as the nutritional value of derived products. Yield and GPC are negatively correlated, and this hampers breeding programs of commercially valuable wheat varieties. The goal of this work was the evaluation of genetic variability for quantity and composition of seed storage proteins, together with yield components [grain yield per spike (GYS) and thousand-kernel weight (TKW)] in a durum wheat population obtained by an inter-specific cross between a common wheat accession and the durum cv. Saragolla. Quantitative trait loci (QTL) analysis was conducted and closely associated markers identified on a genetic map composed of 4,366 SNP markers previously obtained in the same durum population genotyped with the 90K iSelect SNP assay. A total of 22 QTL were detected for traits related to durum wheat quality. Six genomic regions responsible for GPC control were mapped on chromosomes 2B, 3A, 4A, 4B, 5B, and 7B, with major QTL on chromosomes 2B, 4A, and 5B. Nine loci were detected for GYS: two on chromosome 5B and 7A and one on chromosomes 2A, 2B, 4A, 4B, 7B, with the strongest QTL on 2B. Eight QTL were identified for TKW, three of which located on chromosome 3A, two on 1B and one on 4B, 5A, and 5B. Only small overlapping was found among QTL for GYS, TKW, and GPC, and increasing alleles coming from both parents on different chromosomes. Good candidate genes were identified in the QTL confidence intervals for GYS and TKW.&amp;lt;/p&amp;gt;;,citation_author=Angelica Giancaspro;,citation_author=Stefania L. Giove;,citation_author=Antonio Blanco;,citation_author=Agata Gadaleta;,citation_publication_date=2019-11;,citation_cover_date=2019-11;,citation_year=2019;,citation_fulltext_html_url=https://www.frontiersin.org/journals/plant-science/articles/10.3389/fpls.2019.01509/full;,citation_doi=10.3389/fpls.2019.01509;,citation_issn=1664-462X;,citation_volume=10;,citation_journal_title=Frontiers in Plant Science;">
<meta name="citation_reference" content="citation_title=Industrial Hemp and Hemp Byproducts as Sustainable Feedstuffs in Livestock Diets;,citation_abstract=Cannabis sativa, more commonly referred to as industrial hemp, has been used for fiber, food, and medicinal purposes for millennia. Temporarily suffering a setback for parts of the 20th and 21st centuries, interest in hemp production has resurged with the recent rollback of legislative restrictions. Most notably, an interest in the nutritional profile of hemp seed has developed based on its rich array of fatty acids and protein. The increase in hemp production has led to a reciprocating increase in the availability of hemp seed and its associated by-products. While anecdotal and historical accounts exist for feeding hemp to livestock, published reports are limited. This is further complicated by a lack of acceptance in many countries for feeding hemp to livestock. Livestock’s ability to consume unusable waste products from various commodities and turn them into consumable goods is a major component of sustainable agriculture. Research indicates that dietary enrichment with hemp has the potential as a valuable source of protein, fat, and fiber to increase the quality and composition of meat, milk, and eggs; however, to what extent remains unknown. This review aims to explore the nutrient profile of the industrial hemp plant and evaluate our current understanding of hemp used in livestock diets.;,citation_author=Kristine Ely;,citation_author=John Fike;,citation_editor=Dinesh Chandra Agrawal;,citation_editor=Rajiv Kumar;,citation_editor=Muralikrishnan Dhanasekaran;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://doi.org/10.1007/978-981-16-8778-5_6;,citation_doi=10.1007/978-981-16-8778-5_6;,citation_isbn=9789811687785;,citation_inbook_title=Cannabis/Hemp for Sustainable Agriculture and Materials;">
<meta name="citation_reference" content="citation_title=Heritability and Associations among Grain Yield and Quality Traits in Quality Protein Maize (QPM) and Non-QPM Hybrids;,citation_abstract=Maize (Zea mays L.) is the main staple cereal food crop cultivated in southern Africa. Interactions between grain yield and biochemical traits can be useful to plant breeders in making informed decisions on the traits to be considered in breeding programs for high grain yield and enhanced quality. The objectives of this study were to estimate the heritability of grain yield and its related traits, as well as quality traits, and determine the association between quality protein maize (QPM) with non-QPM crosses. Grain yield, and agronomic and quality trait data were obtained from 13 field trials in two countries, for two consecutive seasons. Significant genotypic and phenotypic correlations were recorded for grain yield with protein content (rG = 0.38; rP = 0.25), and tryptophan with oil content (rG = 0.58; rP = 0.25), and negative rG and rP correlations were found for protein with tryptophan content and grain yield with tryptophan content. Path analysis identified ear aspect, ears per plant, and starch as the major traits contributing to grain yield. It is recommended that ear aspect should be considered a key secondary trait in breeding for QPM hybrids. The negative association between grain yield and tryptophan, and between protein and tryptophan, will make it difficult to develop hybrids with high grain yield and high tryptophan content. Hence, it is recommended that gene pyramiding should be considered for these traits.;,citation_author=Isaac Kodzo Amegbor;,citation_author=Angeline Biljon;,citation_author=Nemera Shargie;,citation_author=Amsal Tarekegne;,citation_author=Maryke T. Labuschagne;,citation_publication_date=2022-03;,citation_cover_date=2022-03;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8951685/;,citation_issue=6;,citation_doi=10.3390/plants11060713;,citation_issn=2223-7747;,citation_pmid=35336595;,citation_volume=11;,citation_journal_title=Plants;">
<meta name="citation_reference" content="citation_title=Direct Spectrophotometric Determination of Fat and Moisture in Meat Products;,citation_abstract=SUMMARY— The near-infrared spectral absorption properties of P-mm-thick samples of meat emulsions were measured by direct spectrophotometric techniques. The resulting spectra are interpreted in terms of absorptions from O-H and C-H stretching vibrations combined with scatter losses. Optical-density differences are correlated with fat and moisture contents. The difference in optical density between 1.80 and 1.725 p gave a high correlation with moisture content and the difference between 1.725 and 1.65 P gave a high correlation with fat content. Direct spectrophotometric analysis predicted fat content within a standard error of ± 2.1% and moisture content within ± 1.4%. The possibilities of this technique are explored and the problems to be solved in developing a rapid, accurate method are discussed;,citation_author=Itamar Ben-Gera;,citation_author=Karl H. Norris;,citation_publication_date=1968;,citation_cover_date=1968;,citation_year=1968;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1365-2621.1968.tb00885.x;,citation_issue=1;,citation_doi=10.1111/j.1365-2621.1968.tb00885.x;,citation_issn=1750-3841;,citation_volume=33;,citation_journal_title=Journal of Food Science;">
<meta name="citation_reference" content="citation_title=Measuring Protein Content in Food: An Overview of Methods;,citation_abstract=In order to determine the quantity of protein in food, it is important to have standardized analytical methods. Several methods exist that are used in different food industries to quantify protein content, including the Kjeldahl, Lowry, Bradford and total amino acid content methods. The correct determination of the protein content of foods is important as, often, as is the case with milk, it determines the economic value of the food product and it can impact the economic feasibility of new industries for alternative protein production. This editorial provides an overview of different protein determination methods and describes their advantages and disadvantages.;,citation_author=Maria Hayes;,citation_publication_date=2020-09;,citation_cover_date=2020-09;,citation_year=2020;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7597951/;,citation_issue=10;,citation_doi=10.3390/foods9101340;,citation_issn=2304-8158;,citation_pmid=32977393;,citation_volume=9;,citation_journal_title=Foods;">
<meta name="citation_reference" content="citation_title=Building Predictive Models in R Using the caret Package;,citation_author=Kuhn;,citation_author=Max;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_fulltext_html_url=https://www.jstatsoft.org/index.php/jss/article/view/v028i05;,citation_issue=5;,citation_doi=10.18637/jss.v028.i05;,citation_volume=28;,citation_journal_title=Journal of Statistical Software;">
<meta name="citation_reference" content="citation_title=Pls: Partial Least Squares and Principal Component Regression;,citation_author=Kristian Hovde Liland;,citation_author=Bjørn-Helge Mevik;,citation_author=Ron Wehrens;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=pls;">
<meta name="citation_reference" content="citation_title=Welcome to the tidyverse;,citation_author=Hadley Wickham;,citation_author=Mara Averick;,citation_author=Jennifer Bryan;,citation_author=Winston Chang;,citation_author=Lucy D’Agostino McGowan;,citation_author=Romain François;,citation_author=Garrett Grolemund;,citation_author=Alex Hayes;,citation_author=Lionel Henry;,citation_author=Jim Hester;,citation_author=Max Kuhn;,citation_author=Thomas Lin Pedersen;,citation_author=Evan Miller;,citation_author=Stephan Milton Bache;,citation_author=Kirill Müller;,citation_author=Jeroen Ooms;,citation_author=David Robinson;,citation_author=Dana Paige Seidel;,citation_author=Vitalie Spinu;,citation_author=Kohske Takahashi;,citation_author=Davis Vaughan;,citation_author=Claus Wilke;,citation_author=Kara Woo;,citation_author=Hiroaki Yutani;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=43;,citation_doi=10.21105/joss.01686;,citation_volume=4;,citation_journal_title=Journal of Open Source Software;">
<meta name="citation_reference" content="citation_title=R: A Language and Environment for Statistical Computing;,citation_author=R Core Team;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://www.R-project.org/;">
<meta name="citation_reference" content="citation_title=Data.table: Extension of ‘data.frame‘;,citation_author=Tyson Barrett;,citation_author=Matt Dowle;,citation_author=Arun Srinivasan;,citation_author=Jan Gorecki;,citation_author=Michael Chirico;,citation_author=Toby Hocking;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=data.table;">
<meta name="citation_reference" content="citation_title=Potential of near- and mid-infrared spectroscopy in biofuel production;,citation_author=James B. Reeves;,citation_publication_date=2012-01-01;,citation_cover_date=2012-01-01;,citation_year=2012;,citation_fulltext_html_url=https://doi.org/10.1080/00103624.2012.641844;,citation_issue=1-2;,citation_doi=10.1080/00103624.2012.641844;,citation_volume=43;,citation_journal_title=Communications in Soil Science and Plant Analysis;">
<meta name="citation_reference" content="citation_title=R: A language and environment for statistical computing;,citation_author=R Core Team;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://www.R-project.org/;">
<meta name="citation_reference" content="citation_title=Building predictive models in r using the caret package;,citation_author=Kuhn;,citation_author=Max;,citation_publication_date=2008;,citation_cover_date=2008;,citation_year=2008;,citation_fulltext_html_url=https://www.jstatsoft.org/index.php/jss/article/view/v028i05;,citation_issue=5;,citation_doi=10.18637/jss.v028.i05;,citation_volume=28;,citation_journal_title=Journal of Statistical Software;">
<meta name="citation_reference" content="citation_title=data.table: Extension of “data.frame”;,citation_author=Tyson Barrett;,citation_author=Matt Dowle;,citation_author=Arun Srinivasan;,citation_author=Jan Gorecki;,citation_author=Michael Chirico;,citation_author=Toby Hocking;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=data.table;">
<meta name="citation_reference" content="citation_title=emmeans: Estimated marginal means, aka least-squares means;,citation_author=Russell V. Lenth;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=emmeans;">
<meta name="citation_reference" content="citation_title=knitr: A general-purpose package for dynamic report generation in r;,citation_author=Yihui Xie;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://yihui.org/knitr/;">
<meta name="citation_reference" content="citation_title=Dynamic documents with R and knitr;,citation_author=Yihui Xie;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_fulltext_html_url=https://yihui.org/knitr/;">
<meta name="citation_reference" content="citation_title=knitr: A comprehensive tool for reproducible research in R;,citation_author=Yihui Xie;,citation_editor=Victoria Stodden;,citation_editor=Friedrich Leisch;,citation_editor=Roger D. Peng;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_inbook_title=Implementing reproducible computational research;">
<meta name="citation_reference" content="citation_title=Fitting linear mixed-effects models using lme4;,citation_author=Douglas Bates;,citation_author=Martin Mächler;,citation_author=Ben Bolker;,citation_author=Steve Walker;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_issue=1;,citation_doi=10.18637/jss.v067.i01;,citation_volume=67;,citation_journal_title=Journal of Statistical Software;">
<meta name="citation_reference" content="citation_title=pls: Partial least squares and principal component regression;,citation_author=Kristian Hovde Liland;,citation_author=Bjørn-Helge Mevik;,citation_author=Ron Wehrens;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=pls;">
<meta name="citation_reference" content="citation_title=An introduction to the prospectr package;,citation_author=Antoine Stevens;,citation_author=Leornardo Ramirez-Lopez;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;">
<meta name="citation_reference" content="citation_title=Classification and regression by randomForest;,citation_author=Andy Liaw;,citation_author=Matthew Wiener;,citation_publication_date=2002;,citation_cover_date=2002;,citation_year=2002;,citation_fulltext_html_url=https://CRAN.R-project.org/doc/Rnews/;,citation_issue=3;,citation_volume=2;,citation_journal_title=R News;">
<meta name="citation_reference" content="citation_title=rmarkdown: Dynamic documents for r;,citation_author=JJ Allaire;,citation_author=Yihui Xie;,citation_author=Christophe Dervieux;,citation_author=Jonathan McPherson;,citation_author=Javier Luraschi;,citation_author=Kevin Ushey;,citation_author=Aron Atkins;,citation_author=Hadley Wickham;,citation_author=Joe Cheng;,citation_author=Winston Chang;,citation_author=Richard Iannone;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://github.com/rstudio/rmarkdown;">
<meta name="citation_reference" content="citation_title=R markdown: The definitive guide;,citation_author=Yihui Xie;,citation_author=J. J. Allaire;,citation_author=Garrett Grolemund;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://bookdown.org/yihui/rmarkdown;,citation_isbn=9781138359338;">
<meta name="citation_reference" content="citation_title=R markdown cookbook;,citation_author=Yihui Xie;,citation_author=Christophe Dervieux;,citation_author=Emily Riederer;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://bookdown.org/yihui/rmarkdown-cookbook;,citation_isbn=9780367563837;">
<meta name="citation_reference" content="citation_title=skimr: Compact and flexible summaries of data;,citation_author=Elin Waring;,citation_author=Michael Quinn;,citation_author=Amelia McNamara;,citation_author=Eduardo Arino de la Rubia;,citation_author=Hao Zhu;,citation_author=Shannon Ellis;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://CRAN.R-project.org/package=skimr;">
<meta name="citation_reference" content="citation_title=Tidymodels: A collection of packages for modeling and machine learning using tidyverse principles.;,citation_author=Max Kuhn;,citation_author=Hadley Wickham;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://www.tidymodels.org;">
<meta name="citation_reference" content="citation_title=Welcome to the tidyverse;,citation_author=Hadley Wickham;,citation_author=Mara Averick;,citation_author=Jennifer Bryan;,citation_author=Winston Chang;,citation_author=Lucy D’Agostino McGowan;,citation_author=Romain François;,citation_author=Garrett Grolemund;,citation_author=Alex Hayes;,citation_author=Lionel Henry;,citation_author=Jim Hester;,citation_author=Max Kuhn;,citation_author=Thomas Lin Pedersen;,citation_author=Evan Miller;,citation_author=Stephan Milton Bache;,citation_author=Kirill Müller;,citation_author=Jeroen Ooms;,citation_author=David Robinson;,citation_author=Dana Paige Seidel;,citation_author=Vitalie Spinu;,citation_author=Kohske Takahashi;,citation_author=Davis Vaughan;,citation_author=Claus Wilke;,citation_author=Kara Woo;,citation_author=Hiroaki Yutani;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=43;,citation_doi=10.21105/joss.01686;,citation_volume=4;,citation_journal_title=Journal of Open Source Software;">
</head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Article Notebook</h6>

            <a href="./index.qmd" class="btn btn-primary quarto-download-embed" download="index.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Near Infra-Red Spectroscopy Predicts Crude Protein in Hemp Grain</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
            <div class="quarto-title-meta-author">
          <div class="quarto-title-meta-heading">Authors</div>
          <div class="quarto-title-meta-heading">Affiliations</div>
          
                <div class="quarto-title-meta-contents">
            <p class="author">Ryan Crawford <a href="mailto:rvc3@cornell.edu" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Cornell University
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Jamie Crawford </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Cornell University
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Lawrence B. Smart </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Cornell AgriTech
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Virginia Moore </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Cornell University
                      </p>
                  </div>
                    </div>
        
        <div class="quarto-title-meta">

                      
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">March 12, 2024</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Hemp, Grain, Spectroscopy</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">INTRODUCTION</a></li>
  <li><a href="#materials-and-methods" id="toc-materials-and-methods" class="nav-link" data-scroll-target="#materials-and-methods">MATERIALS AND METHODS</a>
  <ul class="collapse">
  <li><a href="#hemp-grain-sample-background" id="toc-hemp-grain-sample-background" class="nav-link" data-scroll-target="#hemp-grain-sample-background">Hemp Grain Sample Background</a></li>
  <li><a href="#spectral-data-collection-and-preprocessing" id="toc-spectral-data-collection-and-preprocessing" class="nav-link" data-scroll-target="#spectral-data-collection-and-preprocessing">Spectral Data Collection and Preprocessing</a></li>
  <li><a href="#laboratory-validation" id="toc-laboratory-validation" class="nav-link" data-scroll-target="#laboratory-validation">Laboratory Validation</a></li>
  <li><a href="#model-development" id="toc-model-development" class="nav-link" data-scroll-target="#model-development">Model Development</a></li>
  <li><a href="#additional-software-used" id="toc-additional-software-used" class="nav-link" data-scroll-target="#additional-software-used">Additional software used</a></li>
  </ul></li>
  <li><a href="#results-and-discussion" id="toc-results-and-discussion" class="nav-link" data-scroll-target="#results-and-discussion">RESULTS AND DISCUSSION</a>
  <ul class="collapse">
  <li><a href="#laboratory-assay-cp-values" id="toc-laboratory-assay-cp-values" class="nav-link" data-scroll-target="#laboratory-assay-cp-values">Laboratory assay CP values</a></li>
  <li><a href="#preprocessing-methods-comparison" id="toc-preprocessing-methods-comparison" class="nav-link" data-scroll-target="#preprocessing-methods-comparison">Preprocessing methods comparison</a></li>
  </ul></li>
  <li><a href="#acknowledgments" id="toc-acknowledgments" class="nav-link" data-scroll-target="#acknowledgments">ACKNOWLEDGMENTS</a></li>
  <li><a href="#supplemental-material" id="toc-supplemental-material" class="nav-link" data-scroll-target="#supplemental-material">SUPPLEMENTAL MATERIAL</a></li>
  <li><a href="#optional-sections" id="toc-optional-sections" class="nav-link" data-scroll-target="#optional-sections">OPTIONAL SECTIONS</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">REFERENCES</a></li>
  <li><a href="#figures-and-tables" id="toc-figures-and-tables" class="nav-link" data-scroll-target="#figures-and-tables">FIGURES AND TABLES</a></li>
  <li><a href="#sec-data-methods" id="toc-sec-data-methods" class="nav-link" data-scroll-target="#sec-data-methods">Data &amp; Methods</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion</a></li>
  <li><a href="#references-1" id="toc-references-1" class="nav-link" data-scroll-target="#references-1">References</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <div class="cell-container"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.0     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'

The following objects are masked from 'package:lubridate':

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year

The following objects are masked from 'package:dplyr':

    between, first, last

The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: lattice

Attaching package: 'caret'

The following object is masked from 'package:purrr':

    lift</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(prospectr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>[34mprospectr version 0.2.7 -- cakes[39m
[34mcheck the package repository at: https://github.com/l-ramirez-lopez/prospectr[39m</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pls)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'pls'

The following object is masked from 'package:prospectr':

    msc

The following object is masked from 'package:caret':

    R2

The following object is masked from 'package:stats':

    loadings</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidymodels)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──
✔ broom        1.0.5      ✔ rsample      1.2.0 
✔ dials        1.2.1      ✔ tune         1.1.2 
✔ infer        1.0.6      ✔ workflows    1.1.4 
✔ modeldata    1.3.0      ✔ workflowsets 1.0.1 
✔ parsnip      1.2.0      ✔ yardstick    1.3.0 
✔ recipes      1.0.10     
── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ data.table::between()    masks dplyr::between()
✖ scales::discard()        masks purrr::discard()
✖ dplyr::filter()          masks stats::filter()
✖ data.table::first()      masks dplyr::first()
✖ recipes::fixed()         masks stringr::fixed()
✖ dplyr::lag()             masks stats::lag()
✖ data.table::last()       masks dplyr::last()
✖ caret::lift()            masks purrr::lift()
✖ yardstick::precision()   masks caret::precision()
✖ yardstick::recall()      masks caret::recall()
✖ yardstick::sensitivity() masks caret::sensitivity()
✖ yardstick::spec()        masks readr::spec()
✖ yardstick::specificity() masks caret::specificity()
✖ recipes::step()          masks stats::step()
✖ data.table::transpose()  masks purrr::transpose()
• Use suppressPackageStartupMessages() to eliminate package startup messages</code></pre>
</div>
</div></div>
<p><strong>incomplete: may contain errors, run-ons, half-thoughts, etc.</strong></p>
<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">INTRODUCTION</h2>
<p>Hemp (Cannabis sativa L.) is an annual crop with potential uses as a source of food or feed from grain, and bast fiber or hurd from the stalk. Hemp cultivars are commonly grown for one or both purposes and a cultivar may be referred to as a grain, fiber, or dual-purpose type. Because of protein’s nutritional importance, the protein content of a grain crop is an prime consideration for researchers, producers, and consumers. Whole hemp grain typically contains approximately 20-30% protein <span class="citation" data-cites="ely_industrial_2022 barta_proteomic_2024">(<a href="#ref-ely_industrial_2022" role="doc-biblioref">Ely and Fike 2022</a>; <a href="#ref-barta_proteomic_2024" role="doc-biblioref">Bárta et al. 2024</a>)</span>. Crude protein (CP) is often used as a proxy for the direct measurement of protein concentration and consists of the multiplication of nitrogen concentration by a conversion factor because measuring nitrogen concentration is relatively easy and cheap via laboratory assay <span class="citation" data-cites="hayes_measuring_2020">(<a href="#ref-hayes_measuring_2020" role="doc-biblioref">Hayes 2020</a>)</span>.</p>
<p>Near-infrared spectroscopy (NIRS) technology is rapid, non-destructive, and cheap, and consists of the measurement of NIR radiation reflected from a sample <span class="citation" data-cites="roberts_near-infrared_2004">(<a href="#ref-roberts_near-infrared_2004" role="doc-biblioref">Roberts, Workman, and Reeves 2004</a>)</span>. NIR spectra from many samples are related to laboratory values for components such as moisture, protein, fat, or fiber <span class="citation" data-cites="roberts_near-infrared_2004">(<a href="#ref-roberts_near-infrared_2004" role="doc-biblioref">Roberts, Workman, and Reeves 2004</a>)</span>. NIRS technology has been used since the 1970’s to assess forage CP <span class="citation" data-cites="reeves_potential_2012 williams_application_1975">(<a href="#ref-reeves_potential_2012" role="doc-biblioref">Reeves 2012a</a>; <a href="#ref-williams_application_1975" role="doc-biblioref">Williams 1975</a>)</span>. A NIRS calibration set often consists of samples from one species grown in many environments encompassing the range of expected values from the analyte or analytes <span class="citation" data-cites="chadalavada_nir_2022">(<a href="#ref-chadalavada_nir_2022" role="doc-biblioref">Chadalavada et al. 2022</a>)</span>. Partial least squares regression (PLSR) is a typical method used in the agricultural and food sciences to relate spectra to analyte <span class="citation" data-cites="roberts_near-infrared_2004">(<a href="#ref-roberts_near-infrared_2004" role="doc-biblioref">Roberts, Workman, and Reeves 2004</a>)</span>. PLSR calculates principal components (PCs) which relate to the dependent variable and summarize the spectra and uses a subset of PCs in order to fit the regression model. PLSR is commonly used in spectroscopy because it tends to work well with highly-correlated spectral data. Typically the number of principal components is chosen via cross-validation to avoid overfitting. <strong>CITES FOR ALL OF THIS</strong></p>
<p>A NIRS-scanned sample of undamaged grain may subsequently be grown, an important consideration for a plant breeder. In wheat and corn, grain protein content has been shown to be heritable <span class="citation" data-cites="giancaspro_genetic_2019 geyer_genetics_2022">(<a href="#ref-giancaspro_genetic_2019" role="doc-biblioref">Giancaspro et al. 2019</a>; <a href="#ref-geyer_genetics_2022" role="doc-biblioref">Geyer, Mohler, and Hartl 2022</a>)</span>. This suggests (at least potentially) that NIRS technology could serve as resource to more rapidly identify high CP hemp germplasm, enabling the delivery of higher CP hemp grain cultivars faster.</p>
<p>For this study, a benchtop NIR spectrometer was used to develop a model to predict CP content based on a data set of hemp grain representing multiple years, locations, and cultivars from grain and dual-purpose hemp types using PLSR.</p>
</section>
<section id="materials-and-methods" class="level2">
<h2 class="anchored" data-anchor-id="materials-and-methods">MATERIALS AND METHODS</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>bckgrnd <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./input_data/simplified_data/background_data_set.csv"</span>) <span class="sc">|&gt;</span> <span class="fu">setDT</span>()</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>spectra <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./input_data/simplified_data/train_test_crude_protein.csv"</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a> bckgrnd[,in_ny<span class="sc">:</span><span class="er">=</span> <span class="fu">ifelse</span>(loc<span class="sc">!=</span><span class="st">"kentucky"</span>, T, F)]</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>bg2 <span class="ot">&lt;-</span> bckgrnd[loc<span class="sc">!=</span><span class="st">"kentucky"</span>]</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co"># extract indices of non-kentucky</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>bg_indices <span class="ot">&lt;-</span> bckgrnd[loc<span class="sc">!=</span><span class="st">"kentucky"</span>, which <span class="ot">=</span> T]</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co"># correct names in bg2--should be h-51, NOT hl-51</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>bg2[cultivar<span class="sc">==</span><span class="st">"hl-51"</span>]<span class="sc">$</span>cultivar <span class="ot">&lt;-</span> <span class="st">"h-51"</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co"># check to see if i did the calc correctly</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>bg2[<span class="fu">grepl</span>(<span class="st">"51"</span>, cultivar),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   temp_id harv_year       loc cultivar       type  in_ny
     &lt;int&gt;     &lt;int&gt;    &lt;char&gt;   &lt;char&gt;     &lt;char&gt; &lt;lgcl&gt;
1:     141      2019 rn041_gen     h-51       dual   TRUE
2:     149      2019       mcg     h-51 multistate   TRUE
3:      15      2021    ithaca     h-51 multistate   TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a> <span class="co">#looks good </span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span>  <span class="fu">table</span>(bckgrnd<span class="sc">$</span>in_ny)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># now take correct spectra filtering out stuff from KY</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>spectra_2 <span class="ot">&lt;-</span> spectra[bg_indices]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="hemp-grain-sample-background" class="level3">
<h3 class="anchored" data-anchor-id="hemp-grain-sample-background">Hemp Grain Sample Background</h3>
<p>Spectral data were obtained from whole (unground) hemp grain samples, harvested at maturity, collected from 2017 - 2021 from 18 cultivar trials in New York (NY) (149 samples). Grain samples were obtained by hand sampling or mechanical harvest and were cleaned of chaff and dried at 30 C for six days in a forced-air dryer. In total, 38 cultivars were represented in the data set. Cultivars were grain or dual-purpose types and included both commercially available and experimental material.</p>
<p>All cultivar trials were planted in randomized complete block design with each cultivar replicated four times. The 2017 data were comprised of samples from the same thirteen cultivars sampled from six NY locations. For those trials, grain was harvested from each plot individually and aggregated by cultivar within each trial. Four subsamples were drawn from each aggregated sample and scanned separately. These spectra were averaged at each 2 nm increment. All remaining samples from 2018-2021 were collected on a per-plot basis. All possible cultivars and possible locations were represented in 2017, but only a selected subset of cultivars and locations were represented in 2018-2021.</p>
</section>
<section id="spectral-data-collection-and-preprocessing" class="level3">
<h3 class="anchored" data-anchor-id="spectral-data-collection-and-preprocessing">Spectral Data Collection and Preprocessing</h3>
<p>A benchtop NIR spectrometer (FOSS/ NIR FOSS/ NIR Systems model 5000) was used to obtain the spectra (FOSS North America, Eden Prairie, MN, USA). Spectra were collected every 2 nm from 1100-2498 nm and the logarithm of reciprocal reflectance was recorded.</p>
<p>WINISI software version 1.02A (Infrasoft International, Port Matilda, PA, USA) was used to average the spectra in 2017, as well as to select samples for laboratory assay. Samples were selected according to their spectral distance from their nearest neighbor within the calibration data set with a cutoff of a distance of 0.6 H, where H is approximately equal to the squared Mahalanobis distance divided by the number of principal components used in the calculation <span class="citation" data-cites="garrido-varo_note_2019">(<a href="#ref-garrido-varo_note_2019" role="doc-biblioref">Garrido-Varo, Garcia-Olmo, and Fearn 2019</a>)</span>. Prior to selection selection, spectra were preprocessed using SNV-detrend with settings 1,4,4,1 for the derivative, gap, smooth, and smooth 2 settings respectively.</p>
</section>
<section id="laboratory-validation" class="level3">
<h3 class="anchored" data-anchor-id="laboratory-validation">Laboratory Validation</h3>
<p>Laboratory assays were performed by Dairy One Forage Laboratory (Ithaca, NY). For those assays, 1mm ground samples were analyzed by combustion using a CN628 or CN928 Carbon/Nitrogen Determinator. Samples from 2017 were aggregated as described above, but the remaining samples were not aggregated.</p>
</section>
<section id="model-development" class="level3">
<h3 class="anchored" data-anchor-id="model-development">Model Development</h3>
<p>Calibration and validations sets were created by dividing the laboratory CP values into tertiles according to their percent CP in order to ensure that the range of CP values was present in both calibration and validation sets. Within each tertile, 75% of the samples were randomly assigned to the calibration set and the remaining 25% were assigned to the validation set. For each calibration set, models were developed in caret using PLSR. In fitting the model, the number of principal components was optimized over a grid search from 1-20. Model performance was evaluated with a bootstrapping method using root mean squared error (RMSE) and R<sup>2</sup> statistics in selecting a final model.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>preproc_key <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./input_data/simplified_data/preprocessing_key.csv"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>preproc_key[,full_name<span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(<span class="st">"Raw Spectra"</span>, <span class="st">"First Derivative"</span>, <span class="st">"Savitzky-Golay"</span>, <span class="st">"Gap-segment derivative"</span>,</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Standard Normal Variate"</span>, <span class="st">"Standard Normal Variate/ Savitzky-Golay"</span>, <span class="st">"SNV-Detrend"</span>, <span class="st">"Multiplicative Scatter Correction"</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Initially several of common spectral preprocessing methods were tested by creating 100 calibration and validation sets as described above. Spectral data from those data sets were transformed by each of the following methods: First Derivative, Gap-segment derivative, Multiplicative Scatter Correction, Raw Spectra, Savitzky-Golay, SNV-Detrend, Standard Normal Variate, Standard Normal Variate/ Savitzky-Golay. For each of these preprocessing methods, models were fit and predictions were made on the corresponding validation set (since there were 8 preprocessing methods, 8 separate models were fit for each of the 100 sets. The relationship between the predicted and actual values of validation set were calculated via RMSE, R<sup>2</sup> and Ratio of Performance to InterQuartile distance (RPIQ). An analysis of variance was performed in order to compare the preprocessing methods according to these metrics considering each data set as a subject, with multiple measurements (preprocessing methods) applied to that subject.</p>
<p>Once the most promising preprocessing method was identified, 1000 more data sets were created and analyzed via that method and performance on the validation sets summarized with RMSE, R<sup>2</sup>, and RPIQ.</p>
</section>
<section id="additional-software-used" class="level3">
<h3 class="anchored" data-anchor-id="additional-software-used">Additional software used</h3>
<!-- Additional analyses were performed using R Statistical Software [@r_core_team_r_2024]. Data were tabulated and summarized using the tidyverse and data.table \[@barrett_datatable_2024; @wickham_welcome_2019\]. PLSR models were constructed using the pls package and validated using caret \[@kuhn_building_2008; @liland_pls_2023\]. -->
<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>grateful<span class="sc">::</span><span class="fu">cite_packages</span>(<span class="at">output =</span> <span class="st">"paragraph"</span>, <span class="at">out.dir =</span> <span class="st">"."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p>We used R version 4.3.3 <span class="citation" data-cites="base">(<a href="#ref-base" role="doc-biblioref">R Core Team 2024</a>)</span> and the following R packages: caret v. 6.0.94 <span class="citation" data-cites="caret">(<a href="#ref-caret" role="doc-biblioref">Kuhn and Max 2008</a>)</span>, data.table v. 1.15.2 <span class="citation" data-cites="datatable">(<a href="#ref-datatable" role="doc-biblioref">Barrett et al. 2024</a>)</span>, emmeans v. 1.10.0 <span class="citation" data-cites="emmeans">(<a href="#ref-emmeans" role="doc-biblioref">Lenth 2024</a>)</span>, knitr v. 1.45 <span class="citation" data-cites="knitr2014 knitr2015 knitr2023">(<a href="#ref-knitr2014" role="doc-biblioref">Xie 2014</a>, <a href="#ref-knitr2015" role="doc-biblioref">2015</a>, <a href="#ref-knitr2023" role="doc-biblioref">2023</a>)</span>, lme4 v. 1.1.35.1 <span class="citation" data-cites="lme4">(<a href="#ref-lme4" role="doc-biblioref">Bates et al. 2015</a>)</span>, pls v. 2.8.3 <span class="citation" data-cites="pls">(<a href="#ref-pls" role="doc-biblioref">Liland, Mevik, and Wehrens 2023</a>)</span>, prospectr v. 0.2.7 <span class="citation" data-cites="prospectr">(<a href="#ref-prospectr" role="doc-biblioref">Stevens and Ramirez-Lopez 2024</a>)</span>, randomForest v. 4.7.1.1 <span class="citation" data-cites="randomForest">(<a href="#ref-randomForest" role="doc-biblioref">Liaw and Wiener 2002</a>)</span>, rmarkdown v. 2.26 <span class="citation" data-cites="rmarkdown2018 rmarkdown2020 rmarkdown2024">(<a href="#ref-rmarkdown2018" role="doc-biblioref">Xie, Allaire, and Grolemund 2018</a>; <a href="#ref-rmarkdown2020" role="doc-biblioref">Xie, Dervieux, and Riederer 2020</a>; <a href="#ref-rmarkdown2024" role="doc-biblioref">Allaire et al. 2024</a>)</span>, skimr v. 2.1.5 <span class="citation" data-cites="skimr">(<a href="#ref-skimr" role="doc-biblioref">Waring et al. 2022</a>)</span>, tidymodels v. 1.1.1 <span class="citation" data-cites="tidymodels">(<a href="#ref-tidymodels" role="doc-biblioref">Kuhn and Wickham 2020</a>)</span>, tidyverse v. 2.0.0 <span class="citation" data-cites="tidyverse">(<a href="#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span>.</p>
</div>
</div></div>
</section>
</section>
<section id="results-and-discussion" class="level2">
<h2 class="anchored" data-anchor-id="results-and-discussion">RESULTS AND DISCUSSION</h2>
<section id="laboratory-assay-cp-values" class="level3">
<h3 class="anchored" data-anchor-id="laboratory-assay-cp-values">Laboratory assay CP values</h3>
<p>Laboratory assay percent CP values are summarized in the following table. These are similar to the range of CP values observed in the literature and this indicating an appropriate basis for a chemometric model.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>my_summary <span class="ot">&lt;-</span> spectra_2<span class="sc">$</span>crude_protein <span class="sc">|&gt;</span> skimr<span class="sc">::</span><span class="fu">skim</span>()<span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">5</span><span class="sc">:</span><span class="dv">11</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(round, <span class="dv">1</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(my_summary) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mean"</span>, <span class="st">"sd"</span>, <span class="st">"minimum"</span>, <span class="st">"first quartile"</span>, <span class="st">"median"</span>, <span class="st">"third quartile"</span>, <span class="st">"maximum"</span>) </span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(my_summary, <span class="at">caption =</span> <span class="st">"Summary of Laboratory Assayed CP values (Values are Percent Dry Matter)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Summary of Laboratory Assayed CP values (Values are Percent Dry Matter)</caption>
<thead>
<tr class="header">
<th style="text-align: right;">mean</th>
<th style="text-align: right;">sd</th>
<th style="text-align: right;">minimum</th>
<th style="text-align: right;">first quartile</th>
<th style="text-align: right;">median</th>
<th style="text-align: right;">third quartile</th>
<th style="text-align: right;">maximum</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">26.1</td>
<td style="text-align: right;">2.5</td>
<td style="text-align: right;">20.8</td>
<td style="text-align: right;">23.9</td>
<td style="text-align: right;">26.4</td>
<td style="text-align: right;">28.2</td>
<td style="text-align: right;">30.8</td>
</tr>
</tbody>
</table>
</div>
</div></div>
</section>
<section id="preprocessing-methods-comparison" class="level3">
<h3 class="anchored" data-anchor-id="preprocessing-methods-comparison">Preprocessing methods comparison</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>multi_metric <span class="ot">&lt;-</span> <span class="fu">metric_set</span>(rmse, rsq, rpiq,rpd)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co"># read data back in to work with it</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>prep_key <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./input_data/simplified_data/preprocessing_key.csv"</span>)</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>sims_key <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./input_data/simplified_data/preprocessing_methods_test.csv"</span>)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>long_form <span class="ot">&lt;-</span> <span class="fu">merge</span>(sims_key, prep_key, <span class="at">all.x =</span> T)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co"># now pull the metrics</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="co"># long_form[, multi_metric(y, value), by = c("id", "preproc")]</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>summaries <span class="ot">&lt;-</span> long_form <span class="sc">|&gt;</span> </span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(id, preproc) <span class="sc">|&gt;</span> </span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">multi_metric</span>(y, value)</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(summaries)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 5
     id preproc          .metric .estimator .estimate
  &lt;int&gt; &lt;chr&gt;            &lt;chr&gt;   &lt;chr&gt;          &lt;dbl&gt;
1     1 first_derivative rmse    standard        1.15
2     1 gap_der          rmse    standard        1.31
3     1 msc              rmse    standard        1.12
4     1 raw              rmse    standard        1.23
5     1 sav_gol          rmse    standard        1.19
6     1 snv              rmse    standard        1.18</code></pre>
</div>
</div></div>
<p>This study is limited in that it represents the creation of one model based upon spectra collected from one machine. NIRS calibrations can be unique to a particular machine, if the machines compared are of the same model <span class="citation" data-cites="reeves2012">(<a href="#ref-reeves2012" role="doc-biblioref">Reeves 2012b</a>)</span>. As well, the calibration and validation sets are relatively small.</p>
<p>This research showed the promise of the use of NIRS in order to make predictions concerning %CP in hemp grain using PLS. Promising preprocessing methods were identified and a model was validated. Further research could refine the model by including more samples or by examining other predictive methods.</p>
</section>
</section>
<section id="acknowledgments" class="level2">
<h2 class="anchored" data-anchor-id="acknowledgments">ACKNOWLEDGMENTS</h2>
</section>
<section id="supplemental-material" class="level2">
<h2 class="anchored" data-anchor-id="supplemental-material">SUPPLEMENTAL MATERIAL</h2>
</section>
<section id="optional-sections" class="level2">
<h2 class="anchored" data-anchor-id="optional-sections">OPTIONAL SECTIONS</h2>
</section>
<section id="references" class="level2">
<h2 class="anchored" data-anchor-id="references">REFERENCES</h2>
</section>
<section id="figures-and-tables" class="level2">
<h2 class="anchored" data-anchor-id="figures-and-tables">FIGURES AND TABLES</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>eruptions <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1492</span>, <span class="dv">1585</span>, <span class="dv">1646</span>, <span class="dv">1677</span>, <span class="dv">1712</span>, <span class="dv">1949</span>, <span class="dv">1971</span>, <span class="dv">2021</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>n_eruptions <span class="ot">&lt;-</span> <span class="fu">length</span>(eruptions)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [8]:</pre></div><div id="cell-fig-timeline" class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>) <span class="sc">+</span> <span class="fl">0.1</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(eruptions, <span class="fu">rep</span>(<span class="dv">0</span>, n_eruptions), </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">pch =</span> <span class="st">"|"</span>, <span class="at">axes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">1</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="fu">box</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-timeline" class="quarto-figure quarto-figure-center quarto-float anchored" alt="An event plot of the years of the last 8 eruptions on La Palma.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-timeline-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="index_files/figure-html/fig-timeline-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: Timeline of recent earthquakes on La Palma"><img src="index_files/figure-html/fig-timeline-1.png" class="img-fluid figure-img" alt="An event plot of the years of the last 8 eruptions on La Palma." width="576"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-timeline-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Timeline of recent earthquakes on La Palma
</figcaption>
</figure>
</div>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>avg_years_between_eruptions <span class="ot">&lt;-</span> <span class="fu">mean</span>(<span class="fu">diff</span>(eruptions[<span class="sc">-</span>n_eruptions]))</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>avg_years_between_eruptions</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 79.83333</code></pre>
</div>
</div></div>
<p>Based on data up to and including 1971, eruptions on La Palma happen every 79.8 years on average.</p>
<p>Studies of the magma systems feeding the volcano, such as <span class="citation" data-cites="marrero2019">(<a href="#ref-marrero2019" role="doc-biblioref"><strong>marrero2019?</strong></a>)</span>, have proposed that there are two main magma reservoirs feeding the Cumbre Vieja volcano; one in the mantle (30-40km depth) which charges and in turn feeds a shallower crustal reservoir (10-20km depth).</p>
<p>Eight eruptions have been recorded since the late 1400s (<a href="#fig-timeline" class="quarto-xref">Figure&nbsp;1</a>).</p>
<p>Data and methods are discussed in <a href="#sec-data-methods" class="quarto-xref">Section&nbsp;9</a>.</p>
<p>Let <span class="math inline">\(x\)</span> denote the number of eruptions in a year. Then, <span class="math inline">\(x\)</span> can be modeled by a Poisson distribution</p>
<p><span id="eq-poisson"><span class="math display">\[
p(x) = \frac{e^{-\lambda} \lambda^{x}}{x !}
\tag{1}\]</span></span></p>
<p>where <span class="math inline">\(\lambda\)</span> is the rate of eruptions per year. Using <a href="#eq-poisson" class="quarto-xref">Equation&nbsp;1</a>, the probability of an eruption in the next <span class="math inline">\(t\)</span> years can be calculated.</p>
<div id="tbl-history" class="quarto-float anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-history-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Recent historic eruptions on La Palma
</figcaption>
<div aria-describedby="tbl-history-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table">
<thead>
<tr class="header">
<th>Name</th>
<th>Year</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Current</td>
<td>2021</td>
</tr>
<tr class="even">
<td>Teneguía</td>
<td>1971</td>
</tr>
<tr class="odd">
<td>Nambroque</td>
<td>1949</td>
</tr>
<tr class="even">
<td>El Charco</td>
<td>1712</td>
</tr>
<tr class="odd">
<td>Volcán San Antonio</td>
<td>1677</td>
</tr>
<tr class="even">
<td>Volcán San Martin</td>
<td>1646</td>
</tr>
<tr class="odd">
<td>Tajuya near El Paso</td>
<td>1585</td>
</tr>
<tr class="even">
<td>Montaña Quemada</td>
<td>1492</td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<p><a href="#tbl-history" class="quarto-xref">Table&nbsp;1</a> summarises the eruptions recorded since the colonization of the islands by Europeans in the late 1400s.</p>
<div id="fig-map" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-map-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/la-palma-map.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: Map of La Palma"><img src="images/la-palma-map.png" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-map-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Map of La Palma
</figcaption>
</figure>
</div>
<p>La Palma is one of the west most islands in the Volcanic Archipelago of the Canary Islands (<a href="#fig-map" class="quarto-xref">Figure&nbsp;2</a>).</p>
<div class="quarto-embed-nb-cell" data-notebook="/home/runner/work/glowing-system/glowing-system/notebooks/explore-earthquakes.qmd" data-notebook-title="Explore Earthquakes" data-notebook-cellid="cell-fig-spatial-plot">
<div class="cell-container"><div class="cell-decorator"><pre>In [10]:</pre></div><div id="cell-fig-spatial-plot" class="cell" data-message.hidden="true" data-warning.hidden="true">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>la_palma <span class="op">|&gt;</span> </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  ggplot(aes(Longitude, Latitude)) <span class="op">+</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  geom_point(aes(color <span class="op">=</span> Magnitude, size <span class="op">=</span> <span class="dv">40</span><span class="op">-</span>`Depth(km)`)) <span class="op">+</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  scale_color_viridis_c(direction <span class="op">=</span> <span class="op">-</span><span class="dv">1</span>) <span class="op">+</span> </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  scale_size(<span class="bu">range</span> <span class="op">=</span> c(<span class="fl">0.5</span>, <span class="dv">2</span>), guide <span class="op">=</span> <span class="st">"none"</span>) <span class="op">+</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  theme_bw()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<div id="fig-spatial-plot" class="quarto-figure quarto-figure-center quarto-float anchored" alt="A scatterplot of earthquake locations plotting latitude against longitude.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-spatial-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="index_files/figure-html/notebooks-explore-earthquakes-fig-spatial-plot-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: Locations of earthquakes on La Palma since 2017"><img src="index_files/figure-html/notebooks-explore-earthquakes-fig-spatial-plot-output-1.png" class="img-fluid figure-img" alt="A scatterplot of earthquake locations plotting latitude against longitude."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-spatial-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Locations of earthquakes on La Palma since 2017
</figcaption>
</figure>
</div>
</div>
</div></div>
</div>
<p><a href="#fig-spatial-plot" class="quarto-xref">Figure&nbsp;3</a> shows the location of recent Earthquakes on La Palma.</p>
</section>
<section id="sec-data-methods" class="level2">
<h2 class="anchored" data-anchor-id="sec-data-methods">Data &amp; Methods</h2>
</section>
<section id="conclusion" class="level2">
<h2 class="anchored" data-anchor-id="conclusion">Conclusion</h2>
</section>
<section id="references-1" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="references-1">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-rmarkdown2024" class="csl-entry" role="listitem">
Allaire, JJ, Yihui Xie, Christophe Dervieux, Jonathan McPherson, Javier Luraschi, Kevin Ushey, Aron Atkins, et al. 2024. <em><span class="nocase">rmarkdown</span>: Dynamic Documents for r</em>. <a href="https://github.com/rstudio/rmarkdown">https://github.com/rstudio/rmarkdown</a>.
</div>
<div id="ref-datatable" class="csl-entry" role="listitem">
Barrett, Tyson, Matt Dowle, Arun Srinivasan, Jan Gorecki, Michael Chirico, and Toby Hocking. 2024. <em><span class="nocase">data.table</span>: Extension of <span>“<span class="nocase">data.frame</span>”</span></em>. <a href="https://CRAN.R-project.org/package=data.table">https://CRAN.R-project.org/package=data.table</a>.
</div>
<div id="ref-barta_proteomic_2024" class="csl-entry" role="listitem">
Bárta, Jan, Pavel Roudnický, Markéta Jarošová, Zbyněk Zdráhal, Adéla Stupková, Veronika Bártová, Zlatuše Krejčová, et al. 2024. <span>“Proteomic <span>Profiles</span> of <span>Whole</span> <span>Seeds</span>, <span>Hulls</span>, and <span>Dehulled</span> <span>Seeds</span> of <span>Two</span> <span>Industrial</span> <span>Hemp</span> (<span>Cannabis</span> Sativa <span>L</span>.) <span>Cultivars</span>.”</span> <em>Plants</em> 13 (1): 111. <a href="https://doi.org/10.3390/plants13010111">https://doi.org/10.3390/plants13010111</a>.
</div>
<div id="ref-lme4" class="csl-entry" role="listitem">
Bates, Douglas, Martin Mächler, Ben Bolker, and Steve Walker. 2015. <span>“Fitting Linear Mixed-Effects Models Using <span class="nocase">lme4</span>.”</span> <em>Journal of Statistical Software</em> 67 (1): 1–48. <a href="https://doi.org/10.18637/jss.v067.i01">https://doi.org/10.18637/jss.v067.i01</a>.
</div>
<div id="ref-chadalavada_nir_2022" class="csl-entry" role="listitem">
Chadalavada, Keerthi, Krithika Anbazhagan, Adama Ndour, Sunita Choudhary, William Palmer, Jamie R Flynn, Srikanth Mallayee, et al. 2022. <span>“<span>NIR</span> <span>Instruments</span> and <span>Prediction</span> <span>Methods</span> for <span>Rapid</span> <span>Access</span> to <span>Grain</span> <span>Protein</span> <span>Content</span> in <span>Multiple</span> <span>Cereals</span>.”</span> <em>Sensors (Basel, Switzerland)</em> 22 (10). <a href="https://doi.org/10.3390/s22103710">https://doi.org/10.3390/s22103710</a>.
</div>
<div id="ref-ely_industrial_2022" class="csl-entry" role="listitem">
Ely, Kristine, and John Fike. 2022. <span>“Industrial <span>Hemp</span> and <span>Hemp</span> <span>Byproducts</span> as <span>Sustainable</span> <span>Feedstuffs</span> in <span>Livestock</span> <span>Diets</span>.”</span> In <em>Cannabis/<span>Hemp</span> for <span>Sustainable</span> <span>Agriculture</span> and <span>Materials</span></em>, edited by Dinesh Chandra Agrawal, Rajiv Kumar, and Muralikrishnan Dhanasekaran, 145–62. Singapore: Springer. <a href="https://doi.org/10.1007/978-981-16-8778-5_6">https://doi.org/10.1007/978-981-16-8778-5_6</a>.
</div>
<div id="ref-garrido-varo_note_2019" class="csl-entry" role="listitem">
Garrido-Varo, A, J Garcia-Olmo, and T Fearn. 2019. <span>“A Note on <span>Mahalanobis</span> and Related Distance Measures in <span>WinISI</span> and <span>The</span> <span>Unscrambler</span>.”</span> <em>Journal of Near Infrared Spectroscopy</em> 27 (4): 253–58. <a href="https://doi.org/10.1177/0967033519848296">https://doi.org/10.1177/0967033519848296</a>.
</div>
<div id="ref-geyer_genetics_2022" class="csl-entry" role="listitem">
Geyer, Manuel, Volker Mohler, and Lorenz Hartl. 2022. <span>“Genetics of the <span>Inverse</span> <span>Relationship</span> Between <span>Grain</span> <span>Yield</span> and <span>Grain</span> <span>Protein</span> <span>Content</span> in <span>Common</span> <span>Wheat</span>.”</span> <em>Plants</em> 11 (16): 2146. <a href="https://doi.org/10.3390/plants11162146">https://doi.org/10.3390/plants11162146</a>.
</div>
<div id="ref-giancaspro_genetic_2019" class="csl-entry" role="listitem">
Giancaspro, Angelica, Stefania L. Giove, Antonio Blanco, and Agata Gadaleta. 2019. <span>“Genetic <span>Variation</span> for <span>Protein</span> <span>Content</span> and <span>Yield</span>-<span>Related</span> <span>Traits</span> in a <span>Durum</span> <span>Population</span> <span>Derived</span> <span>From</span> an <span>Inter</span>-<span>Specific</span> <span>Cross</span> <span>Between</span> <span>Hexaploid</span> and <span>Tetraploid</span> <span>Wheat</span> <span>Cultivars</span>.”</span> <em>Frontiers in Plant Science</em> 10 (November). <a href="https://doi.org/10.3389/fpls.2019.01509">https://doi.org/10.3389/fpls.2019.01509</a>.
</div>
<div id="ref-hayes_measuring_2020" class="csl-entry" role="listitem">
Hayes, Maria. 2020. <span>“Measuring <span>Protein</span> <span>Content</span> in <span>Food</span>: <span>An</span> <span>Overview</span> of <span>Methods</span>.”</span> <em>Foods</em> 9 (10): 1340. <a href="https://doi.org/10.3390/foods9101340">https://doi.org/10.3390/foods9101340</a>.
</div>
<div id="ref-tidymodels" class="csl-entry" role="listitem">
Kuhn, Max, and Hadley Wickham. 2020. <em><span>Tidymodels</span>: A Collection of Packages for Modeling and Machine Learning Using Tidyverse Principles.</em> <a href="https://www.tidymodels.org">https://www.tidymodels.org</a>.
</div>
<div id="ref-caret" class="csl-entry" role="listitem">
Kuhn, and Max. 2008. <span>“Building Predictive Models in r Using the Caret Package.”</span> <em>Journal of Statistical Software</em> 28 (5): 1–26. <a href="https://doi.org/10.18637/jss.v028.i05">https://doi.org/10.18637/jss.v028.i05</a>.
</div>
<div id="ref-emmeans" class="csl-entry" role="listitem">
Lenth, Russell V. 2024. <em><span class="nocase">emmeans</span>: Estimated Marginal Means, Aka Least-Squares Means</em>. <a href="https://CRAN.R-project.org/package=emmeans">https://CRAN.R-project.org/package=emmeans</a>.
</div>
<div id="ref-randomForest" class="csl-entry" role="listitem">
Liaw, Andy, and Matthew Wiener. 2002. <span>“Classification and Regression by randomForest.”</span> <em>R News</em> 2 (3): 18–22. <a href="https://CRAN.R-project.org/doc/Rnews/">https://CRAN.R-project.org/doc/Rnews/</a>.
</div>
<div id="ref-pls" class="csl-entry" role="listitem">
Liland, Kristian Hovde, Bjørn-Helge Mevik, and Ron Wehrens. 2023. <em><span class="nocase">pls</span>: Partial Least Squares and Principal Component Regression</em>. <a href="https://CRAN.R-project.org/package=pls">https://CRAN.R-project.org/package=pls</a>.
</div>
<div id="ref-base" class="csl-entry" role="listitem">
R Core Team. 2024. <em><span>R</span>: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-reeves_potential_2012" class="csl-entry" role="listitem">
Reeves, James B. 2012a. <span>“Potential of <span>Near</span>- and <span>Mid</span>-Infrared <span>Spectroscopy</span> in <span>Biofuel</span> <span>Production</span>.”</span> <em>Communications in Soil Science and Plant Analysis</em> 43 (1-2): 478–95. <a href="https://doi.org/10.1080/00103624.2012.641844">https://doi.org/10.1080/00103624.2012.641844</a>.
</div>
<div id="ref-reeves2012" class="csl-entry" role="listitem">
———. 2012b. <span>“Potential of Near- and Mid-Infrared Spectroscopy in Biofuel Production.”</span> <em>Communications in Soil Science and Plant Analysis</em> 43 (1-2): 478–95. <a href="https://doi.org/10.1080/00103624.2012.641844">https://doi.org/10.1080/00103624.2012.641844</a>.
</div>
<div id="ref-roberts_near-infrared_2004" class="csl-entry" role="listitem">
Roberts, Craig A, Jerry Workman, and James B Reeves. 2004. <em>Near-Infrared Spectroscopy in Agriculture</em>. American Society of Agronomy.
</div>
<div id="ref-prospectr" class="csl-entry" role="listitem">
Stevens, Antoine, and Leornardo Ramirez-Lopez. 2024. <em>An Introduction to the Prospectr Package</em>.
</div>
<div id="ref-skimr" class="csl-entry" role="listitem">
Waring, Elin, Michael Quinn, Amelia McNamara, Eduardo Arino de la Rubia, Hao Zhu, and Shannon Ellis. 2022. <em><span class="nocase">skimr</span>: Compact and Flexible Summaries of Data</em>. <a href="https://CRAN.R-project.org/package=skimr">https://CRAN.R-project.org/package=skimr</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome to the <span class="nocase">tidyverse</span>.”</span> <em>Journal of Open Source Software</em> 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
<div id="ref-williams_application_1975" class="csl-entry" role="listitem">
Williams, P. C. 1975. <span>“Application of Near Infrared Reflectance Spectroscopy to Analysis of Cereal Grains and Oilseeds.”</span> <em>Cereal Chemistry</em> 52 (4 p.561-576): 576–61.
</div>
<div id="ref-knitr2014" class="csl-entry" role="listitem">
Xie, Yihui. 2014. <span>“<span class="nocase">knitr</span>: A Comprehensive Tool for Reproducible Research in <span>R</span>.”</span> In <em>Implementing Reproducible Computational Research</em>, edited by Victoria Stodden, Friedrich Leisch, and Roger D. Peng. Chapman; Hall/CRC.
</div>
<div id="ref-knitr2015" class="csl-entry" role="listitem">
———. 2015. <em>Dynamic Documents with <span>R</span> and Knitr</em>. 2nd ed. Boca Raton, Florida: Chapman; Hall/CRC. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-knitr2023" class="csl-entry" role="listitem">
———. 2023. <em><span class="nocase">knitr</span>: A General-Purpose Package for Dynamic Report Generation in r</em>. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-rmarkdown2018" class="csl-entry" role="listitem">
Xie, Yihui, J. J. Allaire, and Garrett Grolemund. 2018. <em>R Markdown: The Definitive Guide</em>. Boca Raton, Florida: Chapman; Hall/CRC. <a href="https://bookdown.org/yihui/rmarkdown">https://bookdown.org/yihui/rmarkdown</a>.
</div>
<div id="ref-rmarkdown2020" class="csl-entry" role="listitem">
Xie, Yihui, Christophe Dervieux, and Emily Riederer. 2020. <em>R Markdown Cookbook</em>. Boca Raton, Florida: Chapman; Hall/CRC. <a href="https://bookdown.org/yihui/rmarkdown-cookbook">https://bookdown.org/yihui/rmarkdown-cookbook</a>.
</div>
</div>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"loop":false,"selector":".lightbox","descPosition":"bottom","closeEffect":"zoom","openEffect":"zoom"});
window.onload = () => {
  lightboxQuarto.on('slide_before_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    const href = trigger.getAttribute('href');
    if (href !== null) {
      const imgEl = window.document.querySelector(`a[href="${href}"] img`);
      if (imgEl !== null) {
        const srcAttr = imgEl.getAttribute("src");
        if (srcAttr && srcAttr.startsWith("data:")) {
          slideConfig.href = srcAttr;
        }
      }
    } 
  });

  lightboxQuarto.on('slide_after_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    if (window.Quarto?.typesetMath) {
      window.Quarto.typesetMath(slideNode);
    }
  });

};
          </script> 
  
</body></html>